03.1-Peve-sRNA-summary
================
Kathleen Durkin
2024-09-05

- <a href="#001-install-and-load-packages"
  id="toc-001-install-and-load-packages">0.0.1 Install and load
  packages</a>
- <a href="#1-srna" id="toc-1-srna">1 sRNA</a>
  - <a href="#11-load-count-data" id="toc-11-load-count-data">1.1 Load count
    data</a>
  - <a href="#12-count-data-munging" id="toc-12-count-data-munging">1.2
    Count data munging</a>
  - <a href="#13-expression-levels" id="toc-13-expression-levels">1.3
    Expression levels</a>
  - <a href="#14-transcript-counts" id="toc-14-transcript-counts">1.4
    Transcript counts</a>
- <a href="#2-mirna" id="toc-2-mirna">2 miRNA</a>
  - <a href="#21-load-mirna-metadata" id="toc-21-load-mirna-metadata">2.1
    Load miRNA metadata</a>
  - <a href="#22-expression-levels" id="toc-22-expression-levels">2.2
    Expression levels</a>
  - <a href="#23-mirna-counts" id="toc-23-mirna-counts">2.3 miRNA counts</a>
  - <a href="#24-heatmap" id="toc-24-heatmap">2.4 Heatmap</a>

Gene expression summary for *Porites evermanni* sRNA-seq data.

- trimmed reads generated in `deep-dive` project

- Reads aligned to *Porites evermanni* transcriptome, details
  [here](https://github.com/urol-e5/deep-dive/blob/main/E-Peve/code/12-Peve-RNAseq-kallisto.md)

### 0.0.1 Install and load packages

``` r
library(tidyverse)
library(ggplot2)
library(reshape2)
library(pheatmap)
library(RColorBrewer)
```

# 1 sRNA

## 1.1 Load count data

Load in the sRNA count matrix generated using ShortStack. Keep in mind
this data includes counts of all sRNAs, not just miRNAs. Also note,
while we have 5 samples from which RNA was sequenced, only 3 of those
were sequenced for sRNA in P. evermanni.

``` r
# Read in sRNA counts data
Peve_counts_sRNA_data_OG <- read_delim("../../../deep-dive/E-Peve/output/08.2-Peve-sRNAseq-ShortStack-31bp-fastp-merged/ShortStack_out/Counts.txt", delim="\t") 
head(Peve_counts_sRNA_data_OG)
```

    # A tibble: 6 × 6
      Coords               Name  MIRNA POR-73-S1-TP2-fastp-…¹ POR-79-S1-TP2-fastp-…²
      <chr>                <chr> <chr>                  <dbl>                  <dbl>
    1 Porites_evermani_sc… Clus… N                          0                     71
    2 Porites_evermani_sc… Clus… N                         58                      0
    3 Porites_evermani_sc… Clus… N                         12                     12
    4 Porites_evermani_sc… Clus… N                        135                     30
    5 Porites_evermani_sc… Clus… N                        113                     35
    6 Porites_evermani_sc… Clus… N                          5                     35
    # ℹ abbreviated names: ¹​`POR-73-S1-TP2-fastp-adapters-polyG-31bp-merged`,
    #   ²​`POR-79-S1-TP2-fastp-adapters-polyG-31bp-merged`
    # ℹ 1 more variable: `POR-82-S1-TP2-fastp-adapters-polyG-31bp-merged` <dbl>

## 1.2 Count data munging

``` r
Peve_counts_sRNA <- Peve_counts_sRNA_data_OG

# Remove excess portions of sample column names to just "sample###"
colnames(Peve_counts_sRNA) <- sub("-S1-TP2-fastp-adapters-polyG-31bp-merged", "", colnames(Peve_counts_sRNA))
colnames(Peve_counts_sRNA) <- sub("POR-", "sample", colnames(Peve_counts_sRNA))

# Keep just the counts and cluster names
Peve_counts_sRNA <- Peve_counts_sRNA %>% select("sample73", "sample79", "sample82", "Name")

# I'm not going to be doing any removal of low-count sRNAs for now

# Make the cluster names our new row names
Peve_counts_sRNA <- Peve_counts_sRNA %>% column_to_rownames(var = "Name")

head(Peve_counts_sRNA)
```

              sample73 sample79 sample82
    Cluster_1        0       71       17
    Cluster_2       58        0        0
    Cluster_3       12       12       26
    Cluster_4      135       30       10
    Cluster_5      113       35       21
    Cluster_6        5       35      129

## 1.3 Expression levels

Plot histograms of the expression levels in each sample

``` r
# Melt the count matrix into long format
Peve_counts_sRNA_melted <- melt(Peve_counts_sRNA, variable.name = "sample", value.name = "counts")

# Plot the expression level histograms for each sample
ggplot(Peve_counts_sRNA_melted, aes(x = counts)) +
  geom_histogram(binwidth = 1, fill = "#1E2761", color = "black") +
  scale_x_log10() +  # Optional: Log-transform the x-axis for better visualization
  facet_wrap(~sample, scales = "free_y") +
  labs(title = "Gene Expression Level Histogram for Each Sample",
       x = "Expression Level (Counts)",
       y = "Frequency") +
  theme_minimal()
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/expression-level-histograms-1.png)<!-- -->

## 1.4 Transcript counts

First let’s check the total number of transcripts in each sample – keep
in mind this expression data has *not* been normalized yet, so there may
be different totals for each sample

``` r
# Calculate the total number of transcripts for each sample
total_transcripts <- colSums(Peve_counts_sRNA)

# Create a data frame for plotting
total_transcripts_df <- data.frame(sample = names(total_transcripts),
                                   totals = total_transcripts)

# Plot the total number of transcripts for each sample
ggplot(total_transcripts_df, aes(x = sample, y = totals)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Transcripts per Sample",
       x = "Sample",
       y = "Total Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/transcript-counts-plot-1.png)<!-- -->

Now let’s check the number of unique transcripts in each sample – that
is, how many unique sRNAs are expressed in each sample? This should be
pretty much the same across samples, even without normalization.

``` r
# Calculate the number of unique transcripts (non-zero counts) for each sample
unique_transcripts <- colSums(Peve_counts_sRNA > 0)

# Create a data frame for plotting
unique_transcripts_df <- data.frame(sample = names(unique_transcripts),
                                    uniques = unique_transcripts)

# Plot the total number of unique transcripts for each sample
ggplot(unique_transcripts_df, aes(x = sample, y = uniques)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Unique Expressed Transcripts per Sample",
       x = "Sample",
       y = "Unique Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/total-unique-transcripts-plot-1.png)<!-- -->

# 2 miRNA

## 2.1 Load miRNA metadata

The ShortStack output Results.txt includes all clusters of sRNA reads,
including those not annotated as valid miRNAs. Now that we’ve looked at
all the sRNAs a bit, let’s focus in on those classified as miRNAs.

``` r
# Join with full metadata sheet, which only contains valid miRNAs
Peve_metadata_miRNA <- read_csv("../../../deep-dive/DEF-cross-species/output/10-shortRNA-ShortStack-comparison/Peve_results_mature_named.csv") 

Peve_counts_sRNA <- rownames_to_column(Peve_counts_sRNA, var = "Name")

Peve_counts_miRNA <- left_join(Peve_metadata_miRNA, Peve_counts_sRNA, by = c("Name" = "Name"))

# Keep just the counts and given miRNA names (e.g., based on match to previously described miRNA)
Peve_counts_miRNA <- Peve_counts_miRNA %>% select("sample73", "sample79", "sample82", "given_miRNA_name")

# We seem to have a duplicate miRNA name, "peve-mir-novel-4", which would prevent us from making the miRNA names our new row names. We need to first make that column unique
Peve_counts_miRNA$given_miRNA_name <- make.unique(as.character(Peve_counts_miRNA$given_miRNA_name))

# Make the miRNA names our new row names
Peve_counts_miRNA <- Peve_counts_miRNA %>% column_to_rownames(var = "given_miRNA_name")

head(Peve_counts_miRNA)
```

                      sample73 sample79 sample82
    peve-mir-100         86625    64520   116691
    peve-mir-novel-31    45912    55736   106369
    peve-mir-novel-3     20505    23687    49156
    peve-mir-novel-21    51104    16437       61
    peve-mir-novel-8     32305     9788    19984
    peve-mir-novel-30    14479    19066    21199

## 2.2 Expression levels

Plot histograms of the miRNA expression levels in each sample

``` r
# Melt the count matrix into long format
Peve_counts_miRNA_melted <- melt(Peve_counts_miRNA, variable.name = "sample", value.name = "counts")

# Plot the expression level histograms for each sample
ggplot(Peve_counts_miRNA_melted, aes(x = counts)) +
  geom_histogram(binwidth = 1, fill = "#1E2761", color = "black") +
  scale_x_log10() +  # Optional: Log-transform the x-axis for better visualization
  facet_wrap(~sample, scales = "free_y") +
  labs(title = "Gene Expression Level Histogram for Each Sample",
       x = "Expression Level (Counts)",
       y = "Frequency") +
  theme_minimal()
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/miRNA-expression-level-histograms-1.png)<!-- -->

## 2.3 miRNA counts

First let’s check the total number of miRNAs in each sample – keep in
mind this expression data has *not* been normalized yet, so there may be
different totals for each sample

``` r
# Calculate the total number of transcripts for each sample
total_miRNA <- colSums(Peve_counts_miRNA)

# Create a data frame for plotting
total_miRNA_df <- data.frame(sample = names(total_miRNA),
                                   totals = total_miRNA)

# Plot the total number of transcripts for each sample
ggplot(total_miRNA_df, aes(x = sample, y = totals)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of miRNAs per Sample",
       x = "Sample",
       y = "Total miRNAs") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/miRNA-counts-plot-1.png)<!-- -->

Now let’s check the number of unique miRNAs in each sample – This should
be pretty much the same across samples, even without normalization.

``` r
# Calculate the number of unique transcripts (non-zero counts) for each sample
unique_miRNA <- colSums(Peve_counts_miRNA > 0)

# Create a data frame for plotting
unique_miRNA_df <- data.frame(sample = names(unique_miRNA),
                                    uniques = unique_miRNA)

# Plot the total number of unique transcripts for each sample
ggplot(unique_miRNA_df, aes(x = sample, y = uniques)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Unique Expressed miRNAs per Sample",
       x = "Sample",
       y = "Unique miRNA") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/total-unique-miRNA-plot-1.png)<!-- -->

## 2.4 Heatmap

``` r
heat_colors <- rev(brewer.pal(12, "RdYlBu"))

pheatmap(Peve_counts_miRNA,
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         show_rownames = TRUE,
         show_colnames = TRUE,
         color = heat_colors,
         fontsize_row = 8,
         fontsize_col = 8)
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/miRNA-heatmap-1.png)<!-- -->
Well… there’s like 3 miRNAs with much higher expression than the others,
which is making visualizing relative differences difficult. Let’s redo
the heatmap, normalizing each row to view relative difference in
expression between samples (`scale='row'`)

``` r
pheatmap(Peve_counts_miRNA,
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         show_rownames = TRUE,
         show_colnames = TRUE,
         scale = 'row',
         color = heat_colors,
         fontsize_row = 8,
         fontsize_col = 8)
```

![](03.1-Peve-sRNA-summary_files/figure-gfm/miRNA-heatmap-rowscale-1.png)<!-- -->
