---
title: "03.2-Peve-lncRNA-summary"
author: "Kathleen Durkin"
date: "2025-05-05"
always_allow_html: true
output: 
  github_document:
    toc: true
    toc_depth: 3
    number_sections: true
    html_preview: true 
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
---

Rerun 7/29/25 with updated lncRNA counts matrix

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = TRUE,         # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```

lncRNA expression summary for *Porites Evermanni* lncRNA, isolated in `18-Peve-lncRNA-matrix`

### Install and load packages

```{r load_libraries, inlcude = TRUE}
library(tidyverse)
library(ggplot2)
library(reshape2)
library(magrittr)
library(DESeq2)
library(RColorBrewer)
library(pheatmap)
```


# Unnormalized data

## Load count data

```{r load-count-data}
# Read in counts data.
Peve_counts_lncRNA_OG <- read.table("../output/18-Peve-lncRNA-matrix/Peve-lncRNA-counts.txt", header=TRUE, skip=1) 
head(Peve_counts_lncRNA_OG)

```

## Count data munging

```{r count-data-munging}
# We need to modify this data frame so that the row names are actually row names, instead of comprising the first column
Peve_counts_lncRNA <- Peve_counts_lncRNA_OG %>%
  column_to_rownames(var = "Geneid")

# Additional formatting
# Remove genomic coordinate info
Peve_counts_lncRNA <- Peve_counts_lncRNA %>% select(-Chr, -Start, -End, -Strand, -Length)

# Remove all transcripts with 5 or fewer counts in all samples
Peve_counts_lncRNA <- Peve_counts_lncRNA[!apply(Peve_counts_lncRNA, 1, function(row) all(row < 6)), ]

# Remove the superfluous portions of the column names, to leave just the sample names
colnames(Peve_counts_lncRNA) <- sub("...data.18.Peve.lncRNA.matrix.RNA.", "", colnames(Peve_counts_lncRNA))
colnames(Peve_counts_lncRNA) <- sub(".sorted.bam", "", colnames(Peve_counts_lncRNA))

# Reorder the columns into alphabetical order (to make it easier to create an associated metadata spreadsheet)
Peve_counts_lncRNA <- Peve_counts_lncRNA[, order(colnames(Peve_counts_lncRNA))]

write.table(Peve_counts_lncRNA, file = "../output/03.2-Peve-lncRNA-summary/Peve_lncRNA_counts_formatted.txt", row.names = TRUE, col.names = TRUE)

Peve_sample_names <- names(Peve_counts_lncRNA)

head(Peve_counts_lncRNA)
Peve_sample_names
```

## Expression levels

Plot histograms of the expression levels in each sample

```{r expression-level-histograms}
# Melt the count matrix into long format
Peve_counts_melted <- melt(Peve_counts_lncRNA, variable.name = "sample", value.name = "counts")

# Plot the expression level histograms for each sample
ggplot(Peve_counts_melted, aes(x = counts)) +
  geom_histogram(binwidth = 1, fill = "#1E2761", color = "black") +
  scale_x_log10() +  # Optional: Log-transform the x-axis for better visualization
  facet_wrap(~sample, scales = "free_y") +
  labs(title = "lncRNA Expression Level Histogram for Each Sample",
       x = "Expression Level (Counts)",
       y = "Frequency") +
  theme_minimal()
```

## Transcript counts

First let's check the total number of transcripts in each sample -- keep in mind this expression data has *not* been normalized yet, so there may be different totals for each sample
```{r transcript-counts-plot}
# Calculate the total number of transcripts for each sample
total_transcripts <- colSums(Peve_counts_lncRNA)

# Create a data frame for plotting
total_transcripts_df <- data.frame(sample = names(total_transcripts),
                                   totals = total_transcripts)

# Plot the total number of transcripts for each sample
ggplot(total_transcripts_df, aes(x = sample, y = totals)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Transcripts per Sample",
       x = "Sample",
       y = "Total Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

Now let's check the number of unique transcripts in each sample -- that is, how many lncRNAs are expressed in each sample? This should be pretty much the same across samples, even without normalization.

```{r total-unique-transcripts-plot}
# Calculate the number of unique transcripts (non-zero counts) for each sample
unique_transcripts <- colSums(Peve_counts_lncRNA > 0)

# Create a data frame for plotting
unique_transcripts_df <- data.frame(sample = names(unique_transcripts),
                                    uniques = unique_transcripts)

# Plot the total number of unique transcripts for each sample
ggplot(unique_transcripts_df, aes(x = sample, y = uniques)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Unique Expressed Transcripts per Sample",
       x = "Sample",
       y = "Unique Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

Some variation, but not as dramatic as in Apul.

# Normalized counts

## Normalize counts with DESeq2

### Plot unnormalized data

```{r plot-unnormalized-RNA}

Peve_counts_lncRNA %>%
  pivot_longer( cols = everything(), names_to = "sample", values_to = "count") %>%
  ggplot(., aes(x = sample, y = count)) +
  geom_violin() + 
  geom_point(alpha = 0.2) +
  theme_minimal() +
  labs(title = "Unnormalized transcript counts",
       x = "Sample",
       y = "count")
```

### Metadata

DESeq2 requires a metadata data frame as input. I don't have sample metadata though so, since we're just doing DESeq2 for normalization purposes (not analysis purposes), I'm just going to create a dummy sheet

```{r make-RNA-metadata-dataframe}
Peve_sample_names <- colnames(Peve_counts_lncRNA)

Peve_metadata_lncRNA <- data.frame(Sample = Peve_sample_names,
                            Species = rep("P.evermanni", 5))
rownames(Peve_metadata_lncRNA) <- Peve_sample_names

head(Peve_metadata_lncRNA)
```

### DESeq object

```{r make-RNA-deseq-object, cache=TRUE}
# Calculate DESeq object
dds_Peve_lncRNA <- DESeqDataSetFromMatrix(countData = Peve_counts_lncRNA,
                              colData = Peve_metadata_lncRNA,
                              design = ~ 1) 

# Run differential expression analysis 
# (Note that this DESeq() function runs all necessary steps, including data normalization, 
# estimating size factors, estimating dispersions, gene-wise dispersion estimates, mean-dispersion 
# relationship, final dispersion estimates, fitting model, and testing)
# Using design = ~1 because we don't have treatment groups

dds_Peve_lncRNA <- DESeq(dds_Peve_lncRNA)
```

It's worth noting here that I'm actually going to be doing two different types of transformation on the counts data, which serve different purposes. 

- First is **normalizing** the transcript counts, which adjusts for differences in library size or sequencing depth, but retains count-like properties. Normalized counts are most useful for things like visualizing expression levels and differential expression analysis.

- Second is **variance stabilizing** the counts data, which aims to make the variance of the transformed data approximately independent of the mean, reducing heteroscedasticity (the relationship between variance and mean) and "smoothing" out the variance at low counts. Notably, the transformed data is *no longer on the original count scale*. The transformation makes the variance roughly constant across the range of counts, which makes it easier to interpret patterns in the data visually. Variance stabilized data is most useful for exploratory data analysis, like PCA, clustering, and heatmaps, and is also the transformation we'll want to use before WGCNA.

```{r transform-RNA-counts, cache=TRUE}
# extract normalized counts
# (normalization is automatically performed by deseq2)
Peve_counts_lncRNA_norm <- counts(dds_Peve_lncRNA, normalized=TRUE) %>% data.frame()

write.table(Peve_counts_lncRNA_norm, file = "../output/03.2-Peve-lncRNA-summary/Peve_counts_lncRNA_normalized.txt", sep = "\t", row.names = TRUE, col.names = TRUE, quote = FALSE)


# variance stabilized data
vsd_Peve_lncRNA <- varianceStabilizingTransformation(dds_Peve_lncRNA, blind=TRUE)
wpn_vsd_Peve_lncRNA <- getVarianceStabilizedData(dds_Peve_lncRNA)
rv_wpn_Peve_lncRNA <- rowVars(wpn_vsd_Peve_lncRNA, useNames=TRUE)

Peve_counts_lncRNA_vsd <- data.frame(wpn_vsd_Peve_lncRNA)
write.table(Peve_counts_lncRNA_vsd, file = "../output/03.2-Peve-lncRNA-summary/Peve_counts_lncRNA_variancestabilized.txt", sep = "\t", row.names = TRUE, col.names = TRUE,quote = FALSE)

q75_wpn_Peve_lncRNA <- quantile(rowVars(wpn_vsd_Peve_lncRNA, useNames=TRUE), .75)  # 75th quantile variability
Peve_counts_lncRNA_vsd_q75 <- wpn_vsd_Peve_lncRNA[ rv_wpn_Peve_lncRNA > q75_wpn_Peve_lncRNA, ] %>% data.frame # filter to retain only the most variable lncRNAs
write.table(Peve_counts_lncRNA_vsd_q75, file = "../output/03.2-Peve-lncRNA-summary/Peve_counts_lncRNA_variancestabilized_q75.txt", sep = "\t", row.names = TRUE, col.names = TRUE,quote = FALSE)

q95_wpn_Peve_lncRNA <- quantile(rowVars(wpn_vsd_Peve_lncRNA, useNames=TRUE), .95)  # 95th quantile variability
Peve_counts_lncRNA_vsd_q95 <- wpn_vsd_Peve_lncRNA[ rv_wpn_Peve_lncRNA > q95_wpn_Peve_lncRNA, ] %>% data.frame # filter to retain only the most variable lncRNAs
write.table(Peve_counts_lncRNA_vsd_q95, file = "../output/03.2-Peve-lncRNA-summary/Peve_counts_lncRNA_variancestabilized_q95.txt", sep = "\t", row.names = TRUE, col.names = TRUE,quote = FALSE)

```

## Plot normalized data

```{r plot-normalized-RNA}
Peve_counts_lncRNA_norm_long <- Peve_counts_lncRNA_norm %>%
  mutate(
    Gene_id = row.names(Peve_counts_lncRNA_norm)
  ) %>%
  pivot_longer(-Gene_id)

Peve_counts_lncRNA_norm_long %>%
  ggplot(., aes(x = name, y = value)) +
  geom_violin() +
  geom_point() +
  theme_bw() +
  theme(
    axis.text.x = element_text( angle = 90)
  ) +
  ylim(0, NA) +
  labs(
    title = "Normalized Expression",
    x = "Sample",
    y = "Normalized counts"
  )
```


## Plot variance stabilized data

```{r plot-vsd-RNA}
Peve_counts_lncRNA_vsd_long <- Peve_counts_lncRNA_vsd %>%
  mutate(
    Gene_id = row.names(Peve_counts_lncRNA_vsd)
  ) %>%
  pivot_longer(-Gene_id)

Peve_counts_lncRNA_vsd_long %>%
  ggplot(., aes(x = name, y = value)) +
  geom_violin() +
  geom_point() +
  theme_bw() +
  theme(
    axis.text.x = element_text( angle = 90)
  ) +
  ylim(0, NA) +
  labs(
    title = "Variance Stabilized Expression",
    x = "Sample",
    y = "Variance stabilized data"
  )
```

## Normalized expression levels

Plot histograms of the normalized expression levels in each sample

```{r norm-expression-level-histograms}
# Melt the count matrix into long format
Peve_counts_norm_melted <- melt(Peve_counts_lncRNA_norm, variable.name = "sample", value.name = "counts")

# Plot the expression level histograms for each sample
ggplot(Peve_counts_norm_melted, aes(x = counts)) +
  geom_histogram(binwidth = 1, fill = "#1E2761", color = "black") +
  scale_x_log10() +  # Optional: Log-transform the x-axis for better visualization
  facet_wrap(~sample, scales = "free_y") +
  labs(title = "lncRNA Expression Level Histogram for Each Sample",
       x = "Expression Level (Counts)",
       y = "Frequency") +
  theme_minimal()
```

## Normalized transcript counts

Check the total number of transcripts in each sample -- now that we've normalized the data these totals should be similar
```{r norm-transcript-counts-plot}
# Calculate the total number of transcripts for each sample
total_transcripts_norm <- colSums(Peve_counts_lncRNA_norm)

# Create a data frame for plotting
total_transcripts_norm_df <- data.frame(sample = names(total_transcripts_norm),
                                   totals = total_transcripts_norm)

# Plot the total number of transcripts for each sample
ggplot(total_transcripts_norm_df, aes(x = sample, y = totals)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Transcripts per Sample",
       x = "Sample",
       y = "Total Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

## PCA of variance stabilized data

```{r PCA}
plotPCA(vsd_Peve_lncRNA, intgroup="Sample")
```

Samples POR-73 and POR-76 are notably divergent from the other samples (and each other), and also have dramatically higher normalized transcript counts than the other 3 samples.

## Sample clustering

```{r sample-clustering}
sample_dists <- dist(t(assay(vsd_Peve_lncRNA)))
pheatmap(as.matrix(sample_dists), clustering_distance_rows = "euclidean", 
         clustering_distance_cols = "euclidean", main="Sample Clustering")
```

## Heatmaps

Of most variable variance stabilized lncRNAs

```{r heatmpas}
# 75th quantile
heat_colors <- rev(brewer.pal(12, "RdYlBu"))
pheatmap(Peve_counts_lncRNA_vsd_q75, 
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         show_rownames = TRUE,
         show_colnames = TRUE,
         color = heat_colors,
         scale="row")

# 95th quantile
pheatmap(Peve_counts_lncRNA_vsd_q95, 
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         show_rownames = TRUE,
         show_colnames = TRUE,
         color = heat_colors,
         scale="row")
```