10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR
================
Kathleen Durkin
2025-06-18

- <a href="#1-mrna-coding-sequence" id="toc-1-mrna-coding-sequence">1 mRNA
  coding sequence</a>
  - <a href="#11-get-mature-mirna-fasta"
    id="toc-11-get-mature-mirna-fasta">1.1 Get mature miRNA fasta</a>
  - <a href="#12-get-coding-sequence-fasta"
    id="toc-12-get-coding-sequence-fasta">1.2 Get coding sequence fasta</a>
  - <a href="#13-run-miranda" id="toc-13-run-miranda">1.3 Run miRanda</a>
  - <a href="#14-summarize-results" id="toc-14-summarize-results">1.4
    Summarize results</a>
- <a href="#2-mirna-and-mrna-5utrs" id="toc-2-mirna-and-mrna-5utrs">2
  miRNA and mRNA 5’UTRs</a>
  - <a href="#21-get-5utr-fasta" id="toc-21-get-5utr-fasta">2.1 Get 5’UTR
    fasta</a>
  - <a href="#22-run-miranda" id="toc-22-run-miranda">2.2 Run miRanda</a>
  - <a href="#23-summarize-results" id="toc-23-summarize-results">2.3
    Summarize results</a>
- <a href="#3-mirna-and-mrna-3utrs" id="toc-3-mirna-and-mrna-3utrs">3
  miRNA and mRNA 3’UTRs</a>
- <a href="#4-summary" id="toc-4-summary">4 Summary</a>
- <a href="#5-combine-with-pcc" id="toc-5-combine-with-pcc">5 Combine with
  PCC</a>
  - <a href="#51-merge-with-cds-miranda-results"
    id="toc-51-merge-with-cds-miranda-results">5.1 Merge with CDS miRanda
    results</a>
  - <a href="#52-merge-with-5utr-miranda-results"
    id="toc-52-merge-with-5utr-miranda-results">5.2 Merge with 5UTR miRanda
    results</a>

``` r
library(dplyr)
```

    ## 
    ## Attaching package: 'dplyr'

    ## The following objects are masked from 'package:stats':
    ## 
    ##     filter, lag

    ## The following objects are masked from 'package:base':
    ## 
    ##     intersect, setdiff, setequal, union

miRanda is a target prediction software, used to identify likely
miRNA-mRNA interactions.

We’ve decided that we need to expand our consideration of potential
miRNA target sites to include the coding sequence and (potentially) the
5’UTR region, since cnidarian miRNAs may function primarily through
target cleavage, instead of translational repression.

Run miRanda using the mRNA coding sequences and the 1kb 5UTR regions as
inputs.

# 1 mRNA coding sequence

## 1.1 Get mature miRNA fasta

``` bash
awk '/^>/ {keep = ($0 ~ /mature/)} keep' ../output/05-Peve-sRNA-ShortStack_4.1.0/ShortStack_out/mir.fasta > ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve_ShortStack_4.1.0_mature.fasta
```

## 1.2 Get coding sequence fasta

``` bash

#/home/shared/bedtools2/bin/bedtools getfasta \
/srlab/programs/bedtools getfasta \
-fi "../data/Porites_evermanni_v1.fa" \
-bed "../output/16-Peve-annotate-UTRs/Peve-genome-mRNA_only.gff" \
-fo "../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve_mRNA_full.fa"
```

## 1.3 Run miRanda

``` bash

# score cutoff >100
# energy cutoff <-10
# strict binding

/home/shared/miRanda-3.3a/src/miranda \
../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve_ShortStack_4.1.0_mature.fasta \
../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve_mRNA_full.fa \
-sc 100 \
-en -20 \
-strict \
-out ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict_all.tab
```

## 1.4 Summarize results

Let’s look at the output

``` bash

echo "miranda run finished!"
echo "counting number of putative interactions predicted"

zgrep -c "Performing Scan" ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict_all.tab

echo "Parsing output"
grep -A 1 "Scores for this hit:" ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict_all.tab | sort | grep '>' > ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt

echo "counting number of putative interactions predicted"
wc -l ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt
```

    ## miranda run finished!
    ## counting number of putative interactions predicted
    ## 423601
    ## Parsing output
    ## counting number of putative interactions predicted
    ## 13934 ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt

This is a lot of putative interactions! Note though, that miRanda only
requires complementarity of a 8bp seed region of the miRNA. We instead
want to look for binding with full or near-full complementarity. Let’s
look at how many putative interactions are predicted for a binding
length of at least 21 nucleotides (the length of our smallest mature
miRNA).

``` bash
echo "number of putative interactions of at least 21 nucleotides"
awk -F'\t' '$7 >= 21' ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt | wc -l
echo ""
echo "check some:"
awk -F'\t' '$7 >= 21' ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt | head -5
```

    ## number of putative interactions of at least 21 nucleotides
    ## 2950
    ## 
    ## check some:
    ## >Cluster_1140.mature::Porites_evermani_scaffold_47:475994-476015(-)  Porites_evermani_scaffold_1158:40952-51791  170.00  -20.37  2 20    6821 6845   21  76.19%  85.71%
    ## >Cluster_1140.mature::Porites_evermani_scaffold_47:475994-476015(-)  Porites_evermani_scaffold_1:220724-266805   175.00  -20.39  2 21    2909 2932   21  80.95%  85.71%
    ## >Cluster_1140.mature::Porites_evermani_scaffold_47:475994-476015(-)  Porites_evermani_scaffold_1:65319-86407 150.00  -20.66  2 21    4146 4175   27  62.96%  70.37%
    ## >Cluster_1140.mature::Porites_evermani_scaffold_47:475994-476015(-)  Porites_evermani_scaffold_165:323644-327106 167.00  -20.89  2 21    934 957 21  76.19%  80.95%
    ## >Cluster_1140.mature::Porites_evermani_scaffold_47:475994-476015(-)  Porites_evermani_scaffold_1963:20398-23043  179.00  -21.62  2 21    2045 2068   21  85.71%  85.71%

We can also see from the percent identities (last 2 entries) that this
number includes alignments with multiple mismatches. Let’s filter again
to reduce the number of permissible mismatches. Let’s say we want no
more than 3 mismatches. For an alignment of 21 nucleotides, this would
be an alignment rate of (21-3)/21 = 85.7%.

``` bash
echo "number of putative interactions of at least 21 nucleotides, with at most 3 mismatches"
awk -F'\t' '$7 >= 21' ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt | awk -F'\t' '$8 >= 85' | wc -l
echo ""
echo "check some:"
awk -F'\t' '$7 >= 21' ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt | awk -F'\t' '$8 >= 85' | head -5
```

    ## number of putative interactions of at least 21 nucleotides, with at most 3 mismatches
    ## 3
    ## 
    ## check some:
    ## >Cluster_1140.mature::Porites_evermani_scaffold_47:475994-476015(-)  Porites_evermani_scaffold_1963:20398-23043  179.00  -21.62  2 21    2045 2068   21  85.71%  85.71%
    ## >Cluster_2882.mature::Porites_evermani_scaffold_148:94583-94604(-)   Porites_evermani_scaffold_2457:23942-47000  179.00  -22.61  2 21    17626 17649 21  85.71%  85.71%
    ## >Cluster_796.mature::Porites_evermani_scaffold_26:382572-382593(-)   Porites_evermani_scaffold_586:46864-50813   178.00  -35.57  2 21    2670 2693   21  85.71%  90.48%

So filtering for full or near-full complementarity reduced the number of
putative interactions between miRNA and full mRNA sequences from 13934
to 3.

# 2 miRNA and mRNA 5’UTRs

We’ve also created a gff of 1kb 5’UTR regions (using the same method
used to define 1kb 3’UTRs, in `16-Peve-annotate-UTRs`). Let’s try
running this through miRanda as well, since it’s possible (though not
necessarily expected) that miRNAs will bind here

## 2.1 Get 5’UTR fasta

``` bash
#/home/shared/bedtools2/bin/bedtools getfasta \
/srlab/programs/bedtools getfasta \
-fi "../data/Porites_evermanni_v1.fa" \
-bed "../output/16-Peve-annotate-UTRs/Peve.GFFannotation.5UTR_1kb_corrected.gff" \
-fo "../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve_5UTR_1kb_corrected.fa"
```

## 2.2 Run miRanda

``` bash

# Same settings we've been using: 
# score cutoff >100
# energy cutoff <-20
# strict binding

/home/shared/miRanda-3.3a/src/miranda \
../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve_ShortStack_4.1.0_mature.fasta \
../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve_5UTR_1kb_corrected.fa \
-sc 100 \
-en -20 \
-strict \
-out ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict_all.tab
```

## 2.3 Summarize results

Let’s look at the output

``` bash

echo "Number of interacting miRNA-lncRNA pairs"
zgrep -c "Performing Scan" ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict_all.tab

echo "Parsing output"
grep -A 1 "Scores for this hit:" ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict_all.tab | sort | grep '>' > ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt

echo "Number of putative interactions predicted"
wc -l ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt
```

    ## Number of interacting miRNA-lncRNA pairs
    ## 1758330
    ## Parsing output
    ## Number of putative interactions predicted
    ## 5054 ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt

How many putative interactions are predicted for a binding length of at
least 21 nucleotides (the length of our smallest mature miRNA)?

``` bash
echo "number of putative interactions:"
wc -l ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt
echo ""
echo "number of putative interactions of at least 21 nucleotides"
awk -F'\t' '$7 >= 21' ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt | wc -l
echo ""
echo "number of putative interactions of at least 21 nucleotides, with at most 3 mismatches"
awk -F'\t' '$7 >= 21' ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt | awk -F'\t' '$8 >= 85' | wc -l
```

    ## number of putative interactions:
    ## 5054 ../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt
    ## 
    ## number of putative interactions of at least 21 nucleotides
    ## 1043
    ## 
    ## number of putative interactions of at least 21 nucleotides, with at most 3 mismatches
    ## 5

So filtering for full or near-full complementarity reduced the number of
putative interactions between miRNA and 5’UTR sequences from 5054 to 5

# 3 miRNA and mRNA 3’UTRs

Now let’s see how filtering changes the outputs of miRanda run with only
the 3’UTR mRNA region (the input we have been using up till now)

``` bash
echo "total number of putative interactions:"
wc -l ../output/10-Peve-mRNA-miRNA-interactions/miranda_strict_all_1kb_parsed_peve_updated.txt

echo "number of putative interactions of at least 21 nucleotides"
awk -F'\t' '$7 >= 21' ../output/10-Peve-mRNA-miRNA-interactions/miranda_strict_all_1kb_parsed_peve_updated.txt | wc -l

echo "number of putative interactions of at least 21 nucleotides, with at most 3 mismatches"
awk -F'\t' '$7 >= 21' ../output/10-Peve-mRNA-miRNA-interactions/miranda_strict_all_1kb_parsed_peve_updated.txt | awk -F'\t' '$8 >= 85' | wc -l
```

    ## total number of putative interactions:
    ## 5067 ../output/10-Peve-mRNA-miRNA-interactions/miranda_strict_all_1kb_parsed_peve_updated.txt
    ## number of putative interactions of at least 21 nucleotides
    ## 1031
    ## number of putative interactions of at least 21 nucleotides, with at most 3 mismatches
    ## 1

When only mRNA 3’UTR regions are used as input, filtering for
full/near-full complementarity reduces the number of putative
interactions from 5067 to 1.

# 4 Summary

| Input     | unfiltered | filtered for complementarity | % retained |
|:----------|:-----------|:-----------------------------|:-----------|
| full mRNA | 13934      | 3                            | 0.0215 %   |
| 5’UTR     | 5054       | 5                            | 0.0989 %   |
| 3’UTR     | 5067       | 1                            | 0.0197 %   |

# 5 Combine with PCC

We’ve also already calculated pairwise Pearson’s correlation
coefficients for every miRNA-mRNA pair. Let’s merge this table with our
CDS and 5UTR miRanda results to see which instances of putative binding
are supported through expression correlation.

PCC table available at
`https://gannet.fish.washington.edu/kdurkin1/ravenbackups/deep-dive-expression/E-Peve/output/10-Peve-mRNA-miRNA-interactions/Peve-PCC_miRNA_mRNA.csv`
(too large to store on Github), originally generated in
`E-Peve/code/10-Peve-mRNA-miRNA-interactions.Rmd`

## 5.1 Merge with CDS miRanda results

Read in PCC and miRanda tables

``` r
# Load
Peve_PCC_miRNA_mRNA <- read.csv("https://gannet.fish.washington.edu/kdurkin1/ravenbackups/deep-dive-expression/E-Peve/output/10-Peve-mRNA-miRNA-interactions/Peve-PCC_miRNA_mRNA.csv") %>% select(-X)
Peve_miRanda_miRNA_CDS <- read.table("../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-mRNA_full-strict-parsed.txt")

# Format the miRNA IDs in miRanda table by removing miRNA coordinates
Peve_miRanda_miRNA_CDS$V1 <- gsub(">", "", Peve_miRanda_miRNA_CDS$V1)
Peve_miRanda_miRNA_CDS$V1 <- sub("\\..*", "", Peve_miRanda_miRNA_CDS$V1)
```

We need to associate the CDS genomic coordinates (used in miRanda
output) with gene IDs (used in PCC output)

``` r
# Load and format mapping table
Peve_mRNA_FUNids <- read.table("../output/16-Peve-annotate-UTRs/Peve-mRNA-geneIDs.txt") %>% select(V1, V4)
Peve_mRNA_FUNids$V4 <- gsub("Name=", "", Peve_mRNA_FUNids$V4)

# Join with miRanda table to annotate genomic coordinates with gene IDs
Peve_miRanda_miRNA_CDS <- left_join(Peve_miRanda_miRNA_CDS, Peve_mRNA_FUNids, by = c("V2" = "V1")) %>% unique()
```

Great! Now we can merge the CDS miRanda results and PCC values

``` r
# Merge
Peve_miRanda_miRNA_CDS_PCC <- left_join(Peve_miRanda_miRNA_CDS, Peve_PCC_miRNA_mRNA, by = c("V1" = "miRNA", "V4.y" = "mRNA")) %>% unique()

# Save
write.csv(Peve_miRanda_miRNA_CDS_PCC, "../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/miRanda_PCC_miRNA_CDS.csv")
```

Note: there may be some NA values in the PCC columns. This would
indicate that one (or both) of the member of that miRNA-mRNA pair had
all-zero counts (i.e. it was unexpressed in all samples)

## 5.2 Merge with 5UTR miRanda results

Read in PCC and miRanda tables

``` r
# Load
Peve_miRanda_miRNA_5UTR <- read.table("../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/Peve-miRanda-5UTR_1kb-strict-parsed.txt")

# Format the miRNA IDs in miRanda table by removing miRNA coordinates
Peve_miRanda_miRNA_5UTR$V1 <- gsub(">", "", Peve_miRanda_miRNA_5UTR$V1)
Peve_miRanda_miRNA_5UTR$V1 <- sub("\\..*", "", Peve_miRanda_miRNA_5UTR$V1)
```

We need to associate the 5UTR genomic coordinates (used in miRanda
output) with gene IDs (used in PCC output)

``` r
# Load and format mapping table
Peve_5UTR_FUNids <- read.table("../output/16-Peve-annotate-UTRs/Peve-5UTR-geneIDs.txt") %>% select(V1, V4)
Peve_5UTR_FUNids$V4 <- gsub("Name=", "", Peve_5UTR_FUNids$V4)

# Join with miRanda table to annotate genomic coordinates with gene IDs
Peve_miRanda_miRNA_5UTR <- left_join(Peve_miRanda_miRNA_5UTR, Peve_5UTR_FUNids, by = c("V2" = "V1")) %>% unique()
```

    ## Warning in left_join(Peve_miRanda_miRNA_5UTR, Peve_5UTR_FUNids, by = c(V2 = "V1")): Detected an unexpected many-to-many relationship between `x` and `y`.
    ## ℹ Row 357 of `x` matches multiple rows in `y`.
    ## ℹ Row 6139 of `y` matches multiple rows in `x`.
    ## ℹ If a many-to-many relationship is expected, set `relationship =
    ##   "many-to-many"` to silence this warning.

Great! Now we can merge the 5UTR miRanda results and PCC values

``` r
# Merge
Peve_miRanda_miRNA_5UTR_PCC <- left_join(Peve_miRanda_miRNA_5UTR, Peve_PCC_miRNA_mRNA, by = c("V1" = "miRNA", "V4.y" = "mRNA")) %>% unique()

# Save
write.csv(Peve_miRanda_miRNA_5UTR_PCC, "../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/miRanda_PCC_miRNA_5UTR.csv")
```

Note: there may be some NA values in the PCC columns. This would
indicate that one (or both) of the member of that miRNA-mRNA pair had
all-zero counts (i.e. it was unexpressed in all samples)
