mRNA-miRNA interactions functional enrichment – 3UTR
================
Kathleen Durkin
2025-07-09

- <a href="#1-format-topgo-files" id="toc-1-format-topgo-files">1 Format
  topGO files</a>
  - <a href="#11-read-in-and-format-annotation-files"
    id="toc-11-read-in-and-format-annotation-files">1.1 Read in and format
    annotation files</a>
  - <a href="#12-set-up-gene2go-object"
    id="toc-12-set-up-gene2go-object">1.2 Set up gene2GO object</a>
  - <a href="#13-define-reference-set" id="toc-13-define-reference-set">1.3
    Define reference set</a>
- <a href="#2-3utr" id="toc-2-3utr">2 3UTR</a>
  - <a href="#21-read-in-pccmiranda-data"
    id="toc-21-read-in-pccmiranda-data">2.1 Read in PCC/miranda data</a>
- <a href="#3-fa-of-all-mirna-targets"
  id="toc-3-fa-of-all-mirna-targets">3 FA of all miRNA targets</a>
- <a href="#4-fe-of-specific-mirnas-targets-all-targets"
  id="toc-4-fe-of-specific-mirnas-targets-all-targets">4 FE of specific
  miRNA’s targets (all targets)</a>
- <a href="#5-fe-of-specific-mirnas-targets-significant-cor-targets"
  id="toc-5-fe-of-specific-mirnas-targets-significant-cor-targets">5 FE of
  specific miRNA’s targets (significant cor targets)</a>
- <a
  href="#6-fe-of-all-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance"
  id="toc-6-fe-of-all-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance">6
  FE of all targets negatively correlated with miRNAs (regardless of
  correlation significance)</a>
  - <a href="#601-biological-processes"
    id="toc-601-biological-processes">6.0.1 Biological Processes</a>
  - <a href="#602-cellular-components"
    id="toc-602-cellular-components">6.0.2 Cellular Components</a>
  - <a href="#603-molecular-functions"
    id="toc-603-molecular-functions">6.0.3 Molecular Functions</a>
  - <a href="#604-join-ontologies" id="toc-604-join-ontologies">6.0.4 Join
    ontologies</a>
- <a
  href="#7-fe-of-all-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance"
  id="toc-7-fe-of-all-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance">7
  FE of all targets positively correlated with miRNAs (regardless of
  correlation significance)</a>
  - <a href="#701-biological-processes"
    id="toc-701-biological-processes">7.0.1 Biological Processes</a>
  - <a href="#702-cellular-components"
    id="toc-702-cellular-components">7.0.2 Cellular Components</a>
  - <a href="#703-molecular-functions"
    id="toc-703-molecular-functions">7.0.3 Molecular Functions</a>
  - <a href="#704-join-ontologies" id="toc-704-join-ontologies">7.0.4 Join
    ontologies</a>
- <a
  href="#8-fe-of-all-targets-significantly-negatively-correlated-with-mirnas"
  id="toc-8-fe-of-all-targets-significantly-negatively-correlated-with-mirnas">8
  FE of all targets significantly negatively correlated with miRNAs</a>
  - <a href="#801-biological-processes"
    id="toc-801-biological-processes">8.0.1 Biological Processes</a>
  - <a href="#802-cellular-components"
    id="toc-802-cellular-components">8.0.2 Cellular Components</a>
  - <a href="#803-molecular-functions"
    id="toc-803-molecular-functions">8.0.3 Molecular Functions</a>
  - <a href="#804-join-ontologies" id="toc-804-join-ontologies">8.0.4 Join
    ontologies</a>
- <a
  href="#9-fe-of-all-targets-significantly-positively-correlated-with-mirnas"
  id="toc-9-fe-of-all-targets-significantly-positively-correlated-with-mirnas">9
  FE of all targets significantly positively correlated with miRNAs</a>
  - <a href="#901-biological-processes"
    id="toc-901-biological-processes">9.0.1 Biological Processes</a>
  - <a href="#902-cellular-components"
    id="toc-902-cellular-components">9.0.2 Cellular Components</a>
  - <a href="#903-molecular-functions"
    id="toc-903-molecular-functions">9.0.3 Molecular Functions</a>
  - <a href="#904-join-ontologies" id="toc-904-join-ontologies">9.0.4 Join
    ontologies</a>

This script will use topGO to analyze functional enrichment of miRNA
targets for Apul in the 3UTR regions

Code used below was created by `Jill Ashey`, modified for use with
P.evermanni datasets by `Kathleen Durkin`

# 1 Format topGO files

## 1.1 Read in and format annotation files

``` r
# Read in Apul annotations
annot <- read.delim("../output/02-Peve-reference-annotation/Porites_evermanni_CDS-IDmapping-2024_09_04.tab")
# Remove unneeded columns 
annot <- annot %>% dplyr::select(-X, -V13)
# Ensure there are no duplicate rows
annot <- annot %>% distinct()
# Rename gene Id column
annot <- dplyr::rename(annot, 
                gene_ID = V1)

# Remove unwanted text from parent column
annot$gene_ID <- gsub("Parent=", "", annot$gene_ID)

head(annot)
```

    ##         gene_ID     V3
    ## 1 Peve_00000001 P61915
    ## 2 Peve_00000002 Q569C3
    ## 3 Peve_00000003 O35654
    ## 4 Peve_00000004 Q96PQ7
    ## 5 Peve_00000005 P52564
    ## 6 Peve_00000006 Q61656
    ##                                                                                                                                                                                                                           Protein.names
    ## 1                                                                                                                                                                     DELTA-actitoxin-Ate1a (DELTA-AITX-Ate1a) (Cytolysin tenebrosin-C)
    ## 2 Ubiquitin carboxyl-terminal hydrolase 1 (EC 3.4.19.12) (Deubiquitinating enzyme 1) (Ubiquitin thioesterase 1) (Ubiquitin-specific-processing protease 1) [Cleaved into: Ubiquitin carboxyl-terminal hydrolase 1, N-terminal fragment]
    ## 3                                                                                                                                                                     DNA polymerase delta subunit 2 (DNA polymerase delta subunit p50)
    ## 4                                                                                                                                                                                                                  Kelch-like protein 5
    ## 5                    Dual specificity mitogen-activated protein kinase kinase 6 (MAP kinase kinase 6) (MAPKK 6) (EC 2.7.12.2) (MAPK/ERK kinase 6) (MEK 6) (Stress-activated protein kinase kinase 3) (SAPK kinase 3) (SAPKK-3) (SAPKK3)
    ## 6                                                                                                 Probable ATP-dependent RNA helicase DDX5 (EC 3.6.4.13) (DEAD box RNA helicase DEAD1) (mDEAD1) (DEAD box protein 5) (RNA helicase p68)
    ##                                                 Organism
    ## 1 Actinia tenebrosa (Australian red waratah sea anemone)
    ## 2                                Rattus norvegicus (Rat)
    ## 3                                   Mus musculus (Mouse)
    ## 4                                   Homo sapiens (Human)
    ## 5                                   Homo sapiens (Human)
    ## 6                                   Mus musculus (Mouse)
    ##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Gene.Ontology..biological.process.
    ## 1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 cation transport [GO:0006812]; cytolysis in another organism [GO:0051715]; pore complex assembly [GO:0046931]
    ## 2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             DNA repair [GO:0006281]; monoubiquitinated protein deubiquitination [GO:0035520]; protein deubiquitination [GO:0016579]; regulation of DNA repair [GO:0006282]; response to UV [GO:0009411]; skeletal system development [GO:0001501]; ubiquitin-dependent protein catabolic process [GO:0006511]
    ## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               DNA biosynthetic process [GO:0071897]; DNA replication [GO:0006260]; DNA strand elongation involved in DNA replication [GO:0006271]; DNA-templated DNA replication [GO:0006261]; error-prone translesion synthesis [GO:0042276]
    ## 4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
    ## 5                                                                                                                                                                        apoptotic process [GO:0006915]; bone development [GO:0060348]; cardiac muscle contraction [GO:0060048]; cellular response to sorbitol [GO:0072709]; cellular senescence [GO:0090398]; MAPK cascade [GO:0000165]; negative regulation of cold-induced thermogenesis [GO:0120163]; nucleotide-binding oligomerization domain containing signaling pathway [GO:0070423]; osteoblast differentiation [GO:0001649]; ovulation cycle process [GO:0022602]; p38MAPK cascade [GO:0038066]; positive regulation of apoptotic process [GO:0043065]; positive regulation of MAP kinase activity [GO:0043406]; positive regulation of nitric-oxide synthase biosynthetic process [GO:0051770]; positive regulation of prostaglandin secretion [GO:0032308]; regulation of cell cycle [GO:0051726]; regulation of signal transduction by p53 class mediator [GO:1901796]; response to ischemia [GO:0002931]; response to xenobiotic stimulus [GO:0009410]; signal transduction [GO:0007165]; signal transduction in response to DNA damage [GO:0042770]; stress-activated MAPK cascade [GO:0051403]
    ## 6 alternative mRNA splicing, via spliceosome [GO:0000380]; androgen receptor signaling pathway [GO:0030521]; BMP signaling pathway [GO:0030509]; circadian rhythm [GO:0007623]; epithelial to mesenchymal transition [GO:0001837]; intracellular estrogen receptor signaling pathway [GO:0030520]; intrinsic apoptotic signaling pathway by p53 class mediator [GO:0072332]; miRNA transcription [GO:0061614]; mRNA transcription [GO:0009299]; myoblast differentiation [GO:0045445]; negative regulation of transcription by RNA polymerase II [GO:0000122]; nuclear-transcribed mRNA catabolic process [GO:0000956]; positive regulation of DNA damage response, signal transduction by p53 class mediator [GO:0043517]; positive regulation of DNA-templated transcription [GO:0045893]; primary miRNA processing [GO:0031053]; protein import into nucleus [GO:0006606]; regulation of alternative mRNA splicing, via spliceosome [GO:0000381]; regulation of androgen receptor signaling pathway [GO:0060765]; regulation of osteoblast differentiation [GO:0045667]; regulation of skeletal muscle cell differentiation [GO:2001014]; regulation of transcription by RNA polymerase II [GO:0006357]; regulation of viral genome replication [GO:0045069]
    ##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Gene.Ontology.IDs
    ## 1                                                                                                                                                                                                                                                                                                                                                                                                                                                             GO:0005576; GO:0006812; GO:0015267; GO:0042151; GO:0044218; GO:0046930; GO:0046931; GO:0051715; GO:0090729
    ## 2                                                                                                                                                                                                                                                                                                                                                                                                             GO:0001501; GO:0004197; GO:0004843; GO:0005634; GO:0005654; GO:0005829; GO:0006281; GO:0006282; GO:0006511; GO:0008233; GO:0009411; GO:0016579; GO:0035520
    ## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                 GO:0003677; GO:0005654; GO:0006260; GO:0006261; GO:0006271; GO:0016035; GO:0042276; GO:0042575; GO:0043625; GO:0071897
    ## 4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         GO:0003779; GO:0005737; GO:0005829; GO:0005856
    ## 5                                                                                                                                     GO:0000165; GO:0001649; GO:0002931; GO:0004674; GO:0004708; GO:0004713; GO:0005524; GO:0005654; GO:0005829; GO:0005856; GO:0006915; GO:0007165; GO:0009410; GO:0019211; GO:0019901; GO:0022602; GO:0032308; GO:0038066; GO:0042770; GO:0042802; GO:0043065; GO:0043406; GO:0043539; GO:0051403; GO:0051726; GO:0051770; GO:0060048; GO:0060348; GO:0070423; GO:0072709; GO:0090398; GO:0106310; GO:0120163; GO:0140537; GO:1901796
    ## 6 GO:0000122; GO:0000380; GO:0000381; GO:0000956; GO:0001837; GO:0003712; GO:0003723; GO:0003724; GO:0003730; GO:0005516; GO:0005524; GO:0005634; GO:0005654; GO:0005730; GO:0005737; GO:0006357; GO:0006606; GO:0007623; GO:0009299; GO:0016607; GO:0016887; GO:0019899; GO:0030509; GO:0030520; GO:0030521; GO:0031053; GO:0035500; GO:0036002; GO:0043021; GO:0043517; GO:0045069; GO:0045445; GO:0045667; GO:0045893; GO:0046332; GO:0048306; GO:0050681; GO:0060765; GO:0061614; GO:0070412; GO:0070878; GO:0071013; GO:0072332; GO:1990841; GO:1990904; GO:2001014

``` r
# Looks good!
```

## 1.2 Set up gene2GO object

Want to isolate a list of GO terms per gene

``` r
gene2go <- annot %>% filter(!is.na(Gene.Ontology.IDs)) %>% dplyr::select(gene_ID, Gene.Ontology.IDs)
gene2go <- gene2go %>% dplyr::rename(GO.ID = Gene.Ontology.IDs)

gene2go_list <- setNames(
  strsplit(as.character(gene2go$GO.ID), ";"), 
  gene2go$gene_ID
)
```

Note: I think this means genes that had a Uniprot ID but no GO terms are
excluded from this analysis

## 1.3 Define reference set

Define reference set of genes. This should be all genes *found in our
samples*, NOT all genes in the P.evermanni genome. Some genes (e.g.,
reproduction pathways) may not be found/expected in our samples for
valid biological reasons.

``` r
# Read in counts matrix
Peve_counts <- read.csv("../output/06-Peve-Hisat/Peve-gene_count_matrix.csv")
# Exclude genes with all 0 counts
Peve_counts <- Peve_counts[rowSums(Peve_counts[, 2:6]) != 0, ]

# Select gene IDs of the genes present in our samples
all_genes <- Peve_counts$gene_id
length(all_genes)
```

    ## [1] 31501

So we have a reference set of 31501 genes present in our samples.

# 2 3UTR

## 2.1 Read in PCC/miranda data

This is a table of all putative miRNA-mRNA binding predicted by miRanda,
plus Pearsons correlation coefficients for coexpression of each putative
binding pair.

``` r
data <- read.csv("../output/10-Peve-mRNA-miRNA-interactions/Peve-miranda_PCC_miRNA_mRNA.csv")

head(data)
```

    ##   X.1    X        miRNA          mRNA    PCC.cor   p_value adjusted_p_value
    ## 1   1  300 Cluster_9149 Peve_00009100 -0.4772517 0.6832660        0.9782621
    ## 2   2  300 Cluster_9149 Peve_00009100 -0.4772517 0.6832660        0.9782621
    ## 3   3  335 Cluster_6914 Peve_00009103  0.4539645 0.7000186        0.9782621
    ## 4   4 1234 Cluster_6906 Peve_00018268 -0.6860633 0.5186746        0.9782621
    ## 5   5 1847  Cluster_589 Peve_00028953 -0.9581593 0.1848079        0.9782621
    ## 6   6 2211 Cluster_2787 Peve_00018488  0.8679339 0.3308953        0.9782621
    ##   score energy query_start_end subject_start_end total_bp_shared query_similar
    ## 1   168 -20.39            2 21           322 343              19        73.68%
    ## 2   168 -20.39            2 21           661 682              19        73.68%
    ## 3   169 -21.38            2 20           302 321              18        83.33%
    ## 4   150 -22.14            2 16             36 60              17        76.47%
    ## 5   152 -23.43            2 13           168 189              11        90.91%
    ## 6   163 -22.50            2 21           241 264              21        71.43%
    ##   subject_similar
    ## 1          84.21%
    ## 2          84.21%
    ## 3          88.89%
    ## 4          76.47%
    ## 5          90.91%
    ## 6          80.95%

Set function to select genes of interest (ie those that have pvalue \<
0.05)

``` r
topDiffGenes <- function(allScore) {
return(allScore < 0.05)}
```

# 3 FA of all miRNA targets

Functional annotation of all putative miRNA targets

``` r
cor_bind_FA <- left_join(data, annot, by = c("mRNA" = "gene_ID")) %>% distinct()

nrow(cor_bind_FA)
```

    ## [1] 3610

``` r
nrow(cor_bind_FA[!is.na(cor_bind_FA$Gene.Ontology.IDs),])
```

    ## [1] 1906

Of the 3610 putative miRNA targets (3UTR region) predicted by miRanda,
1906 have available annotations

``` r
sig_cor_bind_FA <- cor_bind_FA[cor_bind_FA$p_value < 0.05,]

# Remove rows where all values are NA
sig_cor_bind_FA <- sig_cor_bind_FA[!apply(is.na(sig_cor_bind_FA), 1, all), ]

nrow(sig_cor_bind_FA)
```

    ## [1] 208

``` r
nrow(sig_cor_bind_FA[!is.na(sig_cor_bind_FA$Gene.Ontology.IDs),])
```

    ## [1] 94

Of the 208 putative 3UTR miRNA targets predicted by miRanda that are
also have significantly correlated expression, 94 have available
annotations.

Save

``` r
write.csv(cor_bind_FA, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_targets_FA.csv")
write.csv(sig_cor_bind_FA, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_sig_cor_targets_FA.csv")
```

# 4 FE of specific miRNA’s targets (all targets)

Create topGO function for use with miRNA names

``` r
miRNA_topGO_FE <- function(miRNA.name, input_interactions) {
  
  #Isolate genes in our input module of interest
  interacting_genes <- input_interactions %>%
    filter(miRNA == miRNA.name) %>%
    pull(mRNA)
  
  if (length(interacting_genes) > 0 && any(all_genes %in% interacting_genes)) {
    # Create factor for all reference genes, where 1 represents module membership and 0 means the gene is not in module of interest
    gene_list <- factor(as.integer(all_genes %in% interacting_genes))
    names(gene_list) <- all_genes
    str(gene_list)
    
    ## Biological Process ##
    # Create topGO object
    GO_BP <- new("topGOdata", 
                ontology = "BP", # Biological Process
                allGenes = gene_list,
                annot = annFUN.gene2GO, 
                gene2GO = gene2go_list,
                geneSel=topDiffGenes)
    
    # Run GO enrichment test
    GO_BP_FE <- runTest(GO_BP, algorithm = "weight01", statistic = "fisher")
    # View the results
    GO_BP_results <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
    # Filter by significant results
    GO_BP_results$Fisher<-as.numeric(GO_BP_results$Fisher)
    GO_BP_results_sig<-GO_BP_results[GO_BP_results$Fisher<0.05,]
    
    
    ## Molecular Function ##
    # Create topGO object
    GO_MF <- new("topGOdata", 
                ontology = "MF", # Molecular Function
                allGenes = gene_list,
                annot = annFUN.gene2GO, 
                gene2GO = gene2go_list,
                geneSel=topDiffGenes)
    
    # Run GO enrichment test
    GO_MF_FE <- runTest(GO_MF, algorithm = "weight01", statistic = "fisher")
    # View the results
    GO_MF_results <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
    # Filter by significant results
    GO_MF_results$Fisher<-as.numeric(GO_MF_results$Fisher)
    GO_MF_results_sig<-GO_MF_results[GO_MF_results$Fisher<0.05,]
  
    # Return
    # Add type column only if results exist
    if (nrow(GO_BP_results_sig) > 0) {
      GO_BP_results_sig$type <- "Biological.Process"
    }
    if (nrow(GO_MF_results_sig) > 0) {
      GO_MF_results_sig$type <- "Molecular.Function"
    }
    GO_results <- rbind(GO_BP_results_sig, GO_MF_results_sig)
    print(GO_results)
  }
}

miRNA_topGO_FE("Cluster_10051", cor_bind_FA)
```

Loop through all miRNA and run functional enrichment on the miRNA’s
targets (all predicted targets)

``` r
interacting_miRNAs <- unique(cor_bind_FA$miRNA) %>% na.omit
results_all_targets <- NULL  # initialize empty df

for(miRNA in interacting_miRNAs) {
  
  # Run topGO enrichment function
  miRNA_results <- miRNA_topGO_FE(miRNA, cor_bind_FA)
  
  # Only keep results if not empty
  if (!is.null(miRNA_results) && nrow(miRNA_results) > 0)  {
    
    # Add the miRNA source column
    miRNA_results$miRNA <- miRNA

    # Bind to the accumulating results data frame
    results_all_targets <- rbind(results_all_targets, miRNA_results)
  }
}
```

    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 98 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  4 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  5 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 9:   6 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 8:   6 nodes to be scored    (59 eliminated genes)

    ## 
    ##   Level 7:   7 nodes to be scored    (61 eliminated genes)

    ## 
    ##   Level 6:   14 nodes to be scored   (79 eliminated genes)

    ## 
    ##   Level 5:   19 nodes to be scored   (106 eliminated genes)

    ## 
    ##   Level 4:   14 nodes to be scored   (115 eliminated genes)

    ## 
    ##   Level 3:   11 nodes to be scored   (181 eliminated genes)

    ## 
    ##   Level 2:   7 nodes to be scored    (215 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (242 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (55 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (55 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (117 eliminated genes)

    ##        GO.ID                             Term Annotated Significant Expected
    ## 1 GO:0050325 tauropine dehydrogenase activity         3           1     0.01
    ## 2 GO:0005506                 iron ion binding         8           1     0.03
    ##   Fisher               type
    ## 1 0.0098 Molecular.Function
    ## 2 0.0260 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (30 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (166 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (209 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 79 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  4 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  4 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 9:   5 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 8:   4 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 7:   5 nodes to be scored    (28 eliminated genes)

    ## 
    ##   Level 6:   11 nodes to be scored   (37 eliminated genes)

    ## 
    ##   Level 5:   15 nodes to be scored   (39 eliminated genes)

    ## 
    ##   Level 4:   12 nodes to be scored   (44 eliminated genes)

    ## 
    ##   Level 3:   9 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (128 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (168 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 28 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 6:   5 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 5:   5 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 4:   4 nodes to be scored    (60 eliminated genes)

    ## 
    ##   Level 3:   6 nodes to be scored    (101 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (135 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (246 eliminated genes)

    ##        GO.ID                                   Term Annotated Significant
    ## 1 GO:0001696                 gastric acid secretion         2           1
    ## 2 GO:0000026 alpha-1,2-mannosyltransferase activity         1           1
    ## 3 GO:0004089         carbonate dehydratase activity         2           1
    ##   Expected Fisher               type
    ## 1     0.02 0.0170 Biological.Process
    ## 2     0.01 0.0082 Molecular.Function
    ## 3     0.02 0.0163 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (69 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 23 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (26 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (99 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (147 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (193 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0001822                                     kidney development         9
    ## 2 GO:0005231 excitatory extracellular ligand-gated ion channel a...         2
    ## 3 GO:0003777                             microtubule motor activity         4
    ##   Significant Expected Fisher               type
    ## 1           1     0.03  0.025 Biological.Process
    ## 2           1     0.01  0.013 Molecular.Function
    ## 3           1     0.03  0.026 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 62 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 12:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 10:  2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 9:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 8:   3 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 7:   5 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 6:   6 nodes to be scored    (42 eliminated genes)

    ## 
    ##   Level 5:   9 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 4:   12 nodes to be scored   (78 eliminated genes)

    ## 
    ##   Level 3:   11 nodes to be scored   (111 eliminated genes)

    ## 
    ##   Level 2:   8 nodes to be scored    (155 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (193 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 37 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (11 eliminated genes)

    ## 
    ##   Level 7:   4 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 6:   6 nodes to be scored    (31 eliminated genes)

    ## 
    ##   Level 5:   8 nodes to be scored    (50 eliminated genes)

    ## 
    ##   Level 4:   4 nodes to be scored    (79 eliminated genes)

    ## 
    ##   Level 3:   6 nodes to be scored    (195 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (216 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (361 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0002002              regulation of angiotensin levels in blood         2
    ## 2 GO:0001867                  complement activation, lectin pathway         4
    ## 3 GO:0000978 RNA polymerase II cis-regulatory region sequence-sp...        11
    ## 4 GO:0001640 adenylate cyclase inhibiting G protein-coupled glut...         1
    ##   Significant Expected Fisher               type
    ## 1           1     0.02 0.0230 Biological.Process
    ## 2           1     0.05 0.0450 Biological.Process
    ## 3           2     0.14 0.0078 Molecular.Function
    ## 4           1     0.01 0.0131 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (10 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (79 eliminated genes)

    ##        GO.ID                                 Term Annotated Significant
    ## 1 GO:0003950 NAD+ ADP-ribosyltransferase activity         2           1
    ##   Expected Fisher               type
    ## 1        0 0.0033 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 72 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 9:   3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 7:   5 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 6:   11 nodes to be scored   (22 eliminated genes)

    ## 
    ##   Level 5:   15 nodes to be scored   (39 eliminated genes)

    ## 
    ##   Level 4:   12 nodes to be scored   (44 eliminated genes)

    ## 
    ##   Level 3:   9 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (128 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (168 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 5 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ##        GO.ID                   Term Annotated Significant Expected Fisher
    ## 1 GO:0001696 gastric acid secretion         2           1     0.01  0.011
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 14 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (74 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (115 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (225 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0000981 DNA-binding transcription factor activity, RNA poly...         4
    ##   Significant Expected Fisher               type
    ## 1           1     0.02   0.02 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (41 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (42 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (76 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (106 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (151 eliminated genes)

    ##        GO.ID                                          Term Annotated
    ## 1 GO:0050233                     pyranose oxidase activity         1
    ## 2 GO:0003978              UDP-glucose 4-epimerase activity         1
    ## 3 GO:0001609 G protein-coupled adenosine receptor activity         3
    ##   Significant Expected Fisher               type
    ## 1           1     0.00 0.0049 Molecular.Function
    ## 2           1     0.00 0.0049 Molecular.Function
    ## 3           1     0.01 0.0147 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 33 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   5 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 5:   7 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (7 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (98 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 16 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (114 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (133 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (230 eliminated genes)

    ##        GO.ID                                       Term Annotated Significant
    ## 1 GO:0002218       activation of innate immune response         2           1
    ## 2 GO:0005007 fibroblast growth factor receptor activity         4           1
    ##   Expected Fisher               type
    ## 1     0.01 0.0057 Biological.Process
    ## 2     0.01 0.0130 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 14 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (7 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (12 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (28 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (118 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0003831 beta-N-acetylglucosaminylglycopeptide beta-1,4-gala...         2
    ## 2 GO:0004508                steroid 17-alpha-monooxygenase activity         4
    ##   Significant Expected Fisher               type
    ## 1           1     0.01 0.0065 Molecular.Function
    ## 2           1     0.01 0.0131 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 17 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (49 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (160 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (187 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (302 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 13 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (13 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (93 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (98 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (157 eliminated genes)

    ##        GO.ID                          Term Annotated Significant Expected
    ## 1 GO:0004140 dephospho-CoA kinase activity         1           1     0.00
    ## 2 GO:0005506              iron ion binding         8           1     0.03
    ##   Fisher               type
    ## 1 0.0033 Molecular.Function
    ## 2 0.0260 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 5 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (69 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID               Term Annotated Significant Expected Fisher
    ## 1 GO:0001822 kidney development         9           1     0.03  0.025
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 23 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 12:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 10:  1 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 9:   2 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (7 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (16 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (47 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (87 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID                           Term Annotated Significant Expected
    ## 1 GO:0000398 mRNA splicing, via spliceosome         6           1     0.02
    ##   Fisher               type
    ## 1  0.017 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 32 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   5 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 5:   6 nodes to be scored    (45 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (46 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (71 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (90 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (107 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 21 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   4 nodes to be scored    (58 eliminated genes)

    ## 
    ##   Level 3:   6 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (219 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (344 eliminated genes)

    ##        GO.ID              Term Annotated Significant Expected Fisher
    ## 1 GO:0001889 liver development         2           1     0.02  0.017
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 11 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (76 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 7 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ##        GO.ID                         Term Annotated Significant Expected Fisher
    ## 1 GO:0000902           cell morphogenesis         3           1     0.02  0.017
    ## 2 GO:0004989 octopamine receptor activity         6           1     0.02  0.020
    ##                 type
    ## 1 Biological.Process
    ## 2 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (13 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (13 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (109 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ##        GO.ID                      Term Annotated Significant Expected Fisher
    ## 1 GO:0004966 galanin receptor activity         5           1     0.01 0.0082
    ##                 type
    ## 1 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 5 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ##        GO.ID                                          Term Annotated
    ## 1 GO:0004993 G protein-coupled serotonin receptor activity         5
    ##   Significant Expected Fisher               type
    ## 1           1     0.02  0.016 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (41 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (42 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 7 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (39 eliminated genes)

    ##        GO.ID                                    Term Annotated Significant
    ## 1 GO:0004508 steroid 17-alpha-monooxygenase activity         4           1
    ##   Expected Fisher               type
    ## 1     0.01 0.0065 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (72 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 25 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 6:   4 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (34 eliminated genes)

    ## 
    ##   Level 4:   4 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (120 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (189 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (248 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0004745 NAD-retinol dehydrogenase activity         1           1     0.01
    ## 2 GO:0004983   neuropeptide Y receptor activity         4           1     0.03
    ##   Fisher               type
    ## 1 0.0065 Molecular.Function
    ## 2 0.0260 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 16 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (25 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0001580 detection of chemical stimulus involved in sensory ...         1
    ##   Significant Expected Fisher               type
    ## 1           1        0 0.0028 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (101 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (108 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID             Term Annotated Significant Expected Fisher
    ## 1 GO:0001764 neuron migration         3           1     0.01 0.0085
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (16 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (122 eliminated genes)

    ##        GO.ID                          Term Annotated Significant Expected
    ## 1 GO:0004222 metalloendopeptidase activity        16           1     0.03
    ##   Fisher               type
    ## 1  0.026 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 31 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 11:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 10:  2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   4 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 4:   7 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (46 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (69 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (87 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (12 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (114 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (133 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (230 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0015969             guanosine tetraphosphate metabolic process         3
    ## 2 GO:0004693 cyclin-dependent protein serine/threonine kinase ac...         1
    ##   Significant Expected Fisher               type
    ## 1           1     0.01 0.0085 Biological.Process
    ## 2           1     0.00 0.0033 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 24 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   4 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 5:   5 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 4:   4 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (102 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (113 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (120 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 41 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 11:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 6:   5 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 5:   7 nodes to be scored    (24 eliminated genes)

    ## 
    ##   Level 4:   9 nodes to be scored    (72 eliminated genes)

    ## 
    ##   Level 3:   7 nodes to be scored    (142 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (256 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (403 eliminated genes)

    ##        GO.ID                                   Term Annotated Significant
    ## 1 GO:0000082  G1/S transition of mitotic cell cycle        25           2
    ## 2 GO:0003993              acid phosphatase activity         1           1
    ## 3 GO:0004970 ionotropic glutamate receptor activity         3           1
    ## 4 GO:0000287                  magnesium ion binding         5           1
    ##   Expected Fisher               type
    ## 1     0.21 0.0140 Biological.Process
    ## 2     0.01 0.0082 Molecular.Function
    ## 3     0.02 0.0244 Molecular.Function
    ## 4     0.04 0.0404 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (101 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (108 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID                                          Term Annotated
    ## 1 GO:0002244 hematopoietic progenitor cell differentiation         2
    ##   Significant Expected Fisher               type
    ## 1           1     0.01 0.0057 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 11 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (20 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (30 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (42 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (51 eliminated genes)

    ##        GO.ID                    Term Annotated Significant Expected Fisher
    ## 1 GO:0005261 cation channel activity        20           1     0.03  0.033
    ##                 type
    ## 1 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 4 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 17 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (24 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (28 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (30 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (169 eliminated genes)

    ##        GO.ID                  Term Annotated Significant Expected Fisher
    ## 1 GO:0003360 brainstem development         4           1     0.01  0.011
    ## 2 GO:0005524           ATP binding        22           1     0.04  0.036
    ##                 type
    ## 1 Biological.Process
    ## 2 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 24 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 7:   4 nodes to be scored    (11 eliminated genes)

    ## 
    ##   Level 6:   4 nodes to be scored    (12 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (31 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (44 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (48 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (51 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0005245                 voltage-gated calcium channel activity         4
    ## 2 GO:0001409 guanine nucleotide transmembrane transporter activi...         6
    ##   Significant Expected Fisher               type
    ## 1           1     0.01  0.013 Molecular.Function
    ## 2           1     0.02  0.020 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (101 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (108 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID             Term Annotated Significant Expected Fisher
    ## 1 GO:0001764 neuron migration         3           1     0.01 0.0085
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (72 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 25 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (24 eliminated genes)

    ## 
    ##   Level 4:   4 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 3:   7 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (70 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (291 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0004252 serine-type endopeptidase activity        14           1     0.05
    ##   Fisher               type
    ## 1  0.045 Molecular.Function

``` r
head(results_all_targets)
```

    ##        GO.ID                                   Term Annotated Significant
    ## 1 GO:0050325       tauropine dehydrogenase activity         3           1
    ## 2 GO:0005506                       iron ion binding         8           1
    ## 3 GO:0001696                 gastric acid secretion         2           1
    ## 4 GO:0000026 alpha-1,2-mannosyltransferase activity         1           1
    ## 5 GO:0004089         carbonate dehydratase activity         2           1
    ## 6 GO:0001822                     kidney development         9           1
    ##   Expected Fisher               type        miRNA
    ## 1     0.01 0.0098 Molecular.Function Cluster_9149
    ## 2     0.03 0.0260 Molecular.Function Cluster_9149
    ## 3     0.02 0.0170 Biological.Process Cluster_6906
    ## 4     0.01 0.0082 Molecular.Function Cluster_6906
    ## 5     0.02 0.0163 Molecular.Function Cluster_6906
    ## 6     0.03 0.0250 Biological.Process  Cluster_589

Save results

``` r
write.csv(results_all_targets, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_all_targets_topGO_FE.csv")
```

# 5 FE of specific miRNA’s targets (significant cor targets)

Loop through all miRNA and run functional enrichment on the miRNA’s
significantly correlated targets

``` r
interacting_miRNAs_sig <- unique(sig_cor_bind_FA$miRNA) %>% na.omit
results_sig_cor_targets <- NULL  # initialize empty df

for(miRNA in interacting_miRNAs_sig) {
  
  # Run topGO enrichment function
  miRNA_results <- miRNA_topGO_FE(miRNA, sig_cor_bind_FA)
  
  # Only keep results if not empty
  if (!is.null(miRNA_results) && nrow(miRNA_results) > 0) {
    
    # Add the miRNA source column
    miRNA_results$miRNA <- miRNA

    # Bind to the accumulating results data frame
    results_sig_cor_targets <- rbind(results_sig_cor_targets, miRNA_results)
  }
}
```

    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (72 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 4 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 20 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (34 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (36 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (112 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (180 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (209 eliminated genes)

    ##        GO.ID                             Term Annotated Significant Expected
    ## 1 GO:0004983 neuropeptide Y receptor activity         4           1     0.01
    ##   Fisher               type
    ## 1  0.013 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 7 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (10 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (79 eliminated genes)

    ##        GO.ID                                   Term Annotated Significant
    ## 1 GO:0000026 alpha-1,2-mannosyltransferase activity         1           1
    ##   Expected Fisher               type
    ## 1        0 0.0016 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (39 eliminated genes)

    ##        GO.ID                      Term Annotated Significant Expected Fisher
    ## 1 GO:0050233 pyranose oxidase activity         1           1        0 0.0016
    ##                 type
    ## 1 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (69 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID               Term Annotated Significant Expected Fisher
    ## 1 GO:0001822 kidney development         9           1     0.03  0.025
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 5 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (73 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 4 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (110 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 7 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (41 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (46 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (79 eliminated genes)

    ##        GO.ID                          Term Annotated Significant Expected
    ## 1 GO:0004140 dephospho-CoA kinase activity         1           1        0
    ##   Fisher               type
    ## 1 0.0016 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (14 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (122 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0004252 serine-type endopeptidase activity        14           1     0.02
    ##   Fisher               type
    ## 1  0.023 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)

``` r
head(results_sig_cor_targets)
```

    ##        GO.ID                                   Term Annotated Significant
    ## 1 GO:0004983       neuropeptide Y receptor activity         4           1
    ## 2 GO:0000026 alpha-1,2-mannosyltransferase activity         1           1
    ## 3 GO:0050233              pyranose oxidase activity         1           1
    ## 4 GO:0001822                     kidney development         9           1
    ## 5 GO:0004140          dephospho-CoA kinase activity         1           1
    ## 6 GO:0004252     serine-type endopeptidase activity        14           1
    ##   Expected Fisher               type         miRNA
    ## 1     0.01 0.0130 Molecular.Function  Cluster_1167
    ## 2     0.00 0.0016 Molecular.Function  Cluster_6906
    ## 3     0.00 0.0016 Molecular.Function Cluster_14999
    ## 4     0.03 0.0250 Biological.Process  Cluster_4079
    ## 5     0.00 0.0016 Molecular.Function  Cluster_4735
    ## 6     0.02 0.0230 Molecular.Function Cluster_10061

Save results

``` r
write.csv(results_sig_cor_targets, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_sig_cor_targets_topGO_FE.csv")
```

# 6 FE of all targets negatively correlated with miRNAs (regardless of correlation significance)

Filter PCC miranda data

``` r
# Filter so that only negative correlations remain 
neg_corr_data <- data %>%
  filter(PCC.cor < 0)
```

Make list of target genes for input to topGO

``` r
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(neg_corr_data$mRNA))
```

``` r
# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs, regardless of
correlation significance.

### 6.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 212 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  4 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  6 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  9 nodes to be scored    (29 eliminated genes)

    ## 
    ##   Level 9:   16 nodes to be scored   (32 eliminated genes)

    ## 
    ##   Level 8:   16 nodes to be scored   (67 eliminated genes)

    ## 
    ##   Level 7:   20 nodes to be scored   (79 eliminated genes)

    ## 
    ##   Level 6:   32 nodes to be scored   (106 eliminated genes)

    ## 
    ##   Level 5:   37 nodes to be scored   (164 eliminated genes)

    ## 
    ##   Level 4:   35 nodes to be scored   (174 eliminated genes)

    ## 
    ##   Level 3:   25 nodes to be scored   (243 eliminated genes)

    ## 
    ##   Level 2:   10 nodes to be scored   (299 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (325 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
neg_cor_gene2go$GO.ID <- trimws(neg_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- neg_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- GO_BP_En_sig_gene %>% distinct()
```

### 6.0.2 Cellular Components

### 6.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 117 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 11:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 10:  3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   7 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 8:   9 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 7:   16 nodes to be scored   (58 eliminated genes)

    ## 
    ##   Level 6:   20 nodes to be scored   (69 eliminated genes)

    ## 
    ##   Level 5:   23 nodes to be scored   (126 eliminated genes)

    ## 
    ##   Level 4:   18 nodes to be scored   (207 eliminated genes)

    ## 
    ##   Level 3:   14 nodes to be scored   (349 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (422 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (541 eliminated genes)

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
neg_cor_gene2go$GO.ID <- trimws(neg_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- neg_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 6.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_neg_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_neg_corr_df`

``` r
test <- GO_neg_corr_df %>%
  inner_join(neg_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_neg_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-29-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_neg_corr_target_enrichment.pdf", plot, width = 20, height = 35, dpi = 300)
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_neg_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "kidney development"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-30-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_top10GO_neg_corr_target_enrichment.pdf", plot, width = 20, height = 25, dpi = 300)
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_top10GO_neg_corr_target_enrichment.png", plot, width = 20, height = 25, dpi = 300)
```

# 7 FE of all targets positively correlated with miRNAs (regardless of correlation significance)

Filter PCC miranda data

``` r
# Filter so that only positive correlations remain 
pos_corr_data <- data %>%
  filter(PCC.cor > 0)

# Make list of target genes for input to topGO
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(pos_corr_data$mRNA))

# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs with positively
correlated expression, regardless of correlation significance.

### 7.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 102 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  4 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  5 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 9:   7 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 8:   6 nodes to be scored    (59 eliminated genes)

    ## 
    ##   Level 7:   7 nodes to be scored    (64 eliminated genes)

    ## 
    ##   Level 6:   15 nodes to be scored   (73 eliminated genes)

    ## 
    ##   Level 5:   21 nodes to be scored   (102 eliminated genes)

    ## 
    ##   Level 4:   16 nodes to be scored   (108 eliminated genes)

    ## 
    ##   Level 3:   11 nodes to be scored   (191 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (251 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (278 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
pos_cor_gene2go$GO.ID <- trimws(pos_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- pos_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- unique(GO_BP_En_sig_gene)
```

### 7.0.2 Cellular Components

### 7.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 108 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   4 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   6 nodes to be scored    (13 eliminated genes)

    ## 
    ##   Level 7:   10 nodes to be scored   (45 eliminated genes)

    ## 
    ##   Level 6:   21 nodes to be scored   (67 eliminated genes)

    ## 
    ##   Level 5:   19 nodes to be scored   (99 eliminated genes)

    ## 
    ##   Level 4:   22 nodes to be scored   (233 eliminated genes)

    ## 
    ##   Level 3:   15 nodes to be scored   (360 eliminated genes)

    ## 
    ##   Level 2:   8 nodes to be scored    (461 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (553 eliminated genes)

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
pos_cor_gene2go$GO.ID <- trimws(pos_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- pos_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 7.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_pos_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_pos_corr_df`

``` r
test <- GO_pos_corr_df %>%
  inner_join(pos_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_pos_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-44-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_pos_corr_target_enrichment.pdf", plot, width = 20, height = 35, dpi = 300)
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_pos_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "angiogenesis"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-45-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_top10GO_pos_corr_target_enrichment.pdf", plot, width = 20, height = 25, dpi = 300)
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_top10GO_pos_corr_target_enrichment.png", plot, width = 20, height = 25, dpi = 300)
```

# 8 FE of all targets significantly negatively correlated with miRNAs

Filter PCC miranda data

``` r
# Filter so that only significant negative correlations remain 
sig_neg_corr_data <- data %>%
  filter(PCC.cor < 0) %>%
  filter(p_value < 0.05)

# Make list of target genes for input to topGO
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(sig_neg_corr_data$mRNA))

# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs with significantly
negatively correlated expression.

### 8.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (69 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_neg_cor_gene2go$GO.ID <- trimws(sig_neg_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- sig_neg_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- unique(GO_BP_En_sig_gene)
```

### 8.0.2 Cellular Components

### 8.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (104 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (149 eliminated genes)

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_neg_cor_gene2go$GO.ID <- trimws(sig_neg_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- sig_neg_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 8.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_sig_neg_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_sig_neg_corr_df`

``` r
test <- GO_sig_neg_corr_df %>%
  inner_join(sig_neg_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_sig_neg_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-59-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_sig_neg_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "kidney development"        "pyranose oxidase activity"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-60-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_top10GO_sig_neg_corr_target_enrichment.pdf", plot, width = 20, height = 25, dpi = 300)
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_top10GO_sig_neg_corr_target_enrichment.png", plot, width = 20, height = 25, dpi = 300)
```

# 9 FE of all targets significantly positively correlated with miRNAs

Filter PCC miranda data

``` r
# Filter so that only significant negative correlations remain 
sig_pos_corr_data <- data %>%
  filter(PCC.cor > 0) %>%
  filter(p_value < 0.05)

# Make list of target genes for input to topGO
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(sig_pos_corr_data$mRNA))

# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs with significantly
positively correlated expression.

### 9.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (72 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_pos_cor_gene2go$GO.ID <- trimws(sig_pos_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- sig_pos_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- unique(GO_BP_En_sig_gene)
```

### 9.0.2 Cellular Components

### 9.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 37 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 6:   6 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 5:   8 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 4:   7 nodes to be scored    (53 eliminated genes)

    ## 
    ##   Level 3:   6 nodes to be scored    (196 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (268 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (361 eliminated genes)

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_pos_cor_gene2go$GO.ID <- trimws(sig_pos_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- sig_pos_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 9.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_sig_pos_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_pos_corr_df`

``` r
test <- GO_sig_pos_corr_df %>%
  inner_join(sig_pos_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_sig_pos_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-74-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_sig_pos_corr_target_enrichment.pdf", plot, width = 20, height = 35, dpi = 300)
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/miRNA_3UTR_topGO_sig_pos_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "alpha-1,2-mannosyltransferase activity"
    ## [2] "dephospho-CoA kinase activity"         
    ## [3] "neuropeptide Y receptor activity"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.13-Peve-mRNA-miRNA-interactions-FE-3UTR_files/figure-gfm/unnamed-chunk-75-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/top10GO_sig_pos_corr_target_enrichment.pdf", last_plot(), width = 20, height = 25, dpi = 300)
ggsave("../output/10.13-Peve-mRNA-miRNA-interactions-FE-3UTR/top10GO_sig_pos_corr_target_enrichment.png", last_plot(), width = 20, height = 25, dpi = 300)
```
