mRNA-miRNA interactions functional enrichment – 5UTR
================
Kathleen Durkin
2025-07-09

- <a href="#1-format-topgo-files" id="toc-1-format-topgo-files">1 Format
  topGO files</a>
  - <a href="#11-read-in-and-format-annotation-files"
    id="toc-11-read-in-and-format-annotation-files">1.1 Read in and format
    annotation files</a>
  - <a href="#12-set-up-gene2go-object"
    id="toc-12-set-up-gene2go-object">1.2 Set up gene2GO object</a>
  - <a href="#13-define-reference-set" id="toc-13-define-reference-set">1.3
    Define reference set</a>
- <a href="#2-5utr" id="toc-2-5utr">2 5UTR</a>
  - <a href="#21-read-in-pccmiranda-data"
    id="toc-21-read-in-pccmiranda-data">2.1 Read in PCC/miranda data</a>
- <a href="#3-fa-of-all-mirna-targets"
  id="toc-3-fa-of-all-mirna-targets">3 FA of all miRNA targets</a>
- <a href="#4-fe-of-specific-mirnas-targets-all-targets"
  id="toc-4-fe-of-specific-mirnas-targets-all-targets">4 FE of specific
  miRNA’s targets (all targets)</a>
- <a href="#5-fe-of-specific-mirnas-targets-significant-cor-targets"
  id="toc-5-fe-of-specific-mirnas-targets-significant-cor-targets">5 FE of
  specific miRNA’s targets (significant cor targets)</a>
- <a
  href="#6-fe-of-all-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance"
  id="toc-6-fe-of-all-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance">6
  FE of all targets negatively correlated with miRNAs (regardless of
  correlation significance)</a>
  - <a href="#601-biological-processes"
    id="toc-601-biological-processes">6.0.1 Biological Processes</a>
  - <a href="#602-cellular-components"
    id="toc-602-cellular-components">6.0.2 Cellular Components</a>
  - <a href="#603-molecular-functions"
    id="toc-603-molecular-functions">6.0.3 Molecular Functions</a>
  - <a href="#604-join-ontologies" id="toc-604-join-ontologies">6.0.4 Join
    ontologies</a>
- <a
  href="#7-fe-of-all-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance"
  id="toc-7-fe-of-all-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance">7
  FE of all targets positively correlated with miRNAs (regardless of
  correlation significance)</a>
  - <a href="#701-biological-processes"
    id="toc-701-biological-processes">7.0.1 Biological Processes</a>
  - <a href="#702-cellular-components"
    id="toc-702-cellular-components">7.0.2 Cellular Components</a>
  - <a href="#703-molecular-functions"
    id="toc-703-molecular-functions">7.0.3 Molecular Functions</a>
  - <a href="#704-join-ontologies" id="toc-704-join-ontologies">7.0.4 Join
    ontologies</a>
- <a
  href="#8-fe-of-all-targets-significantly-negatively-correlated-with-mirnas"
  id="toc-8-fe-of-all-targets-significantly-negatively-correlated-with-mirnas">8
  FE of all targets significantly negatively correlated with miRNAs</a>
  - <a href="#801-biological-processes"
    id="toc-801-biological-processes">8.0.1 Biological Processes</a>
  - <a href="#802-cellular-components"
    id="toc-802-cellular-components">8.0.2 Cellular Components</a>
  - <a href="#803-molecular-functions"
    id="toc-803-molecular-functions">8.0.3 Molecular Functions</a>
  - <a href="#804-join-ontologies" id="toc-804-join-ontologies">8.0.4 Join
    ontologies</a>
- <a
  href="#9-fe-of-all-targets-significantly-positively-correlated-with-mirnas"
  id="toc-9-fe-of-all-targets-significantly-positively-correlated-with-mirnas">9
  FE of all targets significantly positively correlated with miRNAs</a>
  - <a href="#901-biological-processes"
    id="toc-901-biological-processes">9.0.1 Biological Processes</a>
  - <a href="#902-cellular-components"
    id="toc-902-cellular-components">9.0.2 Cellular Components</a>
  - <a href="#903-molecular-functions"
    id="toc-903-molecular-functions">9.0.3 Molecular Functions</a>
  - <a href="#904-join-ontologies" id="toc-904-join-ontologies">9.0.4 Join
    ontologies</a>

This script will use topGO to analyze functional enrichment of miRNA
targets for Apul in the 5UTR regions

Code used below was created by `Jill Ashey`, modified for use with
P.evermanni datasets by `Kathleen Durkin`

# 1 Format topGO files

## 1.1 Read in and format annotation files

``` r
# Read in Apul annotations
annot <- read.delim("../output/02-Peve-reference-annotation/Porites_evermanni_CDS-IDmapping-2024_09_04.tab")
# Remove unneeded columns 
annot <- annot %>% dplyr::select(-X, -V13)
# Ensure there are no duplicate rows
annot <- annot %>% distinct()
# Rename gene Id column
annot <- dplyr::rename(annot, 
                gene_ID = V1)

# Remove unwanted text from parent column
annot$gene_ID <- gsub("Parent=", "", annot$gene_ID)

head(annot)
```

    ##         gene_ID     V3
    ## 1 Peve_00000001 P61915
    ## 2 Peve_00000002 Q569C3
    ## 3 Peve_00000003 O35654
    ## 4 Peve_00000004 Q96PQ7
    ## 5 Peve_00000005 P52564
    ## 6 Peve_00000006 Q61656
    ##                                                                                                                                                                                                                           Protein.names
    ## 1                                                                                                                                                                     DELTA-actitoxin-Ate1a (DELTA-AITX-Ate1a) (Cytolysin tenebrosin-C)
    ## 2 Ubiquitin carboxyl-terminal hydrolase 1 (EC 3.4.19.12) (Deubiquitinating enzyme 1) (Ubiquitin thioesterase 1) (Ubiquitin-specific-processing protease 1) [Cleaved into: Ubiquitin carboxyl-terminal hydrolase 1, N-terminal fragment]
    ## 3                                                                                                                                                                     DNA polymerase delta subunit 2 (DNA polymerase delta subunit p50)
    ## 4                                                                                                                                                                                                                  Kelch-like protein 5
    ## 5                    Dual specificity mitogen-activated protein kinase kinase 6 (MAP kinase kinase 6) (MAPKK 6) (EC 2.7.12.2) (MAPK/ERK kinase 6) (MEK 6) (Stress-activated protein kinase kinase 3) (SAPK kinase 3) (SAPKK-3) (SAPKK3)
    ## 6                                                                                                 Probable ATP-dependent RNA helicase DDX5 (EC 3.6.4.13) (DEAD box RNA helicase DEAD1) (mDEAD1) (DEAD box protein 5) (RNA helicase p68)
    ##                                                 Organism
    ## 1 Actinia tenebrosa (Australian red waratah sea anemone)
    ## 2                                Rattus norvegicus (Rat)
    ## 3                                   Mus musculus (Mouse)
    ## 4                                   Homo sapiens (Human)
    ## 5                                   Homo sapiens (Human)
    ## 6                                   Mus musculus (Mouse)
    ##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Gene.Ontology..biological.process.
    ## 1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 cation transport [GO:0006812]; cytolysis in another organism [GO:0051715]; pore complex assembly [GO:0046931]
    ## 2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             DNA repair [GO:0006281]; monoubiquitinated protein deubiquitination [GO:0035520]; protein deubiquitination [GO:0016579]; regulation of DNA repair [GO:0006282]; response to UV [GO:0009411]; skeletal system development [GO:0001501]; ubiquitin-dependent protein catabolic process [GO:0006511]
    ## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               DNA biosynthetic process [GO:0071897]; DNA replication [GO:0006260]; DNA strand elongation involved in DNA replication [GO:0006271]; DNA-templated DNA replication [GO:0006261]; error-prone translesion synthesis [GO:0042276]
    ## 4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
    ## 5                                                                                                                                                                        apoptotic process [GO:0006915]; bone development [GO:0060348]; cardiac muscle contraction [GO:0060048]; cellular response to sorbitol [GO:0072709]; cellular senescence [GO:0090398]; MAPK cascade [GO:0000165]; negative regulation of cold-induced thermogenesis [GO:0120163]; nucleotide-binding oligomerization domain containing signaling pathway [GO:0070423]; osteoblast differentiation [GO:0001649]; ovulation cycle process [GO:0022602]; p38MAPK cascade [GO:0038066]; positive regulation of apoptotic process [GO:0043065]; positive regulation of MAP kinase activity [GO:0043406]; positive regulation of nitric-oxide synthase biosynthetic process [GO:0051770]; positive regulation of prostaglandin secretion [GO:0032308]; regulation of cell cycle [GO:0051726]; regulation of signal transduction by p53 class mediator [GO:1901796]; response to ischemia [GO:0002931]; response to xenobiotic stimulus [GO:0009410]; signal transduction [GO:0007165]; signal transduction in response to DNA damage [GO:0042770]; stress-activated MAPK cascade [GO:0051403]
    ## 6 alternative mRNA splicing, via spliceosome [GO:0000380]; androgen receptor signaling pathway [GO:0030521]; BMP signaling pathway [GO:0030509]; circadian rhythm [GO:0007623]; epithelial to mesenchymal transition [GO:0001837]; intracellular estrogen receptor signaling pathway [GO:0030520]; intrinsic apoptotic signaling pathway by p53 class mediator [GO:0072332]; miRNA transcription [GO:0061614]; mRNA transcription [GO:0009299]; myoblast differentiation [GO:0045445]; negative regulation of transcription by RNA polymerase II [GO:0000122]; nuclear-transcribed mRNA catabolic process [GO:0000956]; positive regulation of DNA damage response, signal transduction by p53 class mediator [GO:0043517]; positive regulation of DNA-templated transcription [GO:0045893]; primary miRNA processing [GO:0031053]; protein import into nucleus [GO:0006606]; regulation of alternative mRNA splicing, via spliceosome [GO:0000381]; regulation of androgen receptor signaling pathway [GO:0060765]; regulation of osteoblast differentiation [GO:0045667]; regulation of skeletal muscle cell differentiation [GO:2001014]; regulation of transcription by RNA polymerase II [GO:0006357]; regulation of viral genome replication [GO:0045069]
    ##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Gene.Ontology.IDs
    ## 1                                                                                                                                                                                                                                                                                                                                                                                                                                                             GO:0005576; GO:0006812; GO:0015267; GO:0042151; GO:0044218; GO:0046930; GO:0046931; GO:0051715; GO:0090729
    ## 2                                                                                                                                                                                                                                                                                                                                                                                                             GO:0001501; GO:0004197; GO:0004843; GO:0005634; GO:0005654; GO:0005829; GO:0006281; GO:0006282; GO:0006511; GO:0008233; GO:0009411; GO:0016579; GO:0035520
    ## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                 GO:0003677; GO:0005654; GO:0006260; GO:0006261; GO:0006271; GO:0016035; GO:0042276; GO:0042575; GO:0043625; GO:0071897
    ## 4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         GO:0003779; GO:0005737; GO:0005829; GO:0005856
    ## 5                                                                                                                                     GO:0000165; GO:0001649; GO:0002931; GO:0004674; GO:0004708; GO:0004713; GO:0005524; GO:0005654; GO:0005829; GO:0005856; GO:0006915; GO:0007165; GO:0009410; GO:0019211; GO:0019901; GO:0022602; GO:0032308; GO:0038066; GO:0042770; GO:0042802; GO:0043065; GO:0043406; GO:0043539; GO:0051403; GO:0051726; GO:0051770; GO:0060048; GO:0060348; GO:0070423; GO:0072709; GO:0090398; GO:0106310; GO:0120163; GO:0140537; GO:1901796
    ## 6 GO:0000122; GO:0000380; GO:0000381; GO:0000956; GO:0001837; GO:0003712; GO:0003723; GO:0003724; GO:0003730; GO:0005516; GO:0005524; GO:0005634; GO:0005654; GO:0005730; GO:0005737; GO:0006357; GO:0006606; GO:0007623; GO:0009299; GO:0016607; GO:0016887; GO:0019899; GO:0030509; GO:0030520; GO:0030521; GO:0031053; GO:0035500; GO:0036002; GO:0043021; GO:0043517; GO:0045069; GO:0045445; GO:0045667; GO:0045893; GO:0046332; GO:0048306; GO:0050681; GO:0060765; GO:0061614; GO:0070412; GO:0070878; GO:0071013; GO:0072332; GO:1990841; GO:1990904; GO:2001014

``` r
# Looks good!
```

## 1.2 Set up gene2GO object

Want to isolate a list of GO terms per gene

``` r
gene2go <- annot %>% filter(!is.na(Gene.Ontology.IDs)) %>% dplyr::select(gene_ID, Gene.Ontology.IDs)
gene2go <- gene2go %>% dplyr::rename(GO.ID = Gene.Ontology.IDs)

gene2go_list <- setNames(
  strsplit(as.character(gene2go$GO.ID), ";"), 
  gene2go$gene_ID
)
```

Note: I think this means genes that had a Uniprot ID but no GO terms are
excluded from this analysis

## 1.3 Define reference set

Define reference set of genes. This should be all genes *found in our
samples*, NOT all genes in the P.evermanni genome. Some genes (e.g.,
reproduction pathways) may not be found/expected in our samples for
valid biological reasons.

``` r
# Read in counts matrix
Peve_counts <- read.csv("../output/06-Peve-Hisat/Peve-gene_count_matrix.csv")
# Exclude genes with all 0 counts
Peve_counts <- Peve_counts[rowSums(Peve_counts[, 2:6]) != 0, ]

# Select gene IDs of the genes present in our samples
all_genes <- Peve_counts$gene_id
length(all_genes)
```

    ## [1] 31501

So we have a reference set of 31501 genes present in our samples.

# 2 5UTR

## 2.1 Read in PCC/miranda data

This is a table of all putative miRNA-mRNA binding predicted by miRanda,
plus Pearsons correlation coefficients for coexpression of each putative
binding pair.

``` r
data <- read.csv("../output/10.01-Peve-mRNA-miRNA-interactions-CDS_5UTR/miRanda_PCC_miRNA_5UTR.csv")
data <- dplyr::rename(data,
               miRNA = V1,
               mRNA = V4.y)
head(data)
```

    ##   X         miRNA                                          V2  V3   V4.x V5 V6
    ## 1 1 Cluster_10060  Porites_evermani_scaffold_1755:94665-95665 176 -22.19  2 18
    ## 2 2 Cluster_10060  Porites_evermani_scaffold_1891:50981-51981 162 -20.99  2 20
    ## 3 3 Cluster_10060   Porites_evermani_scaffold_1:919565-920565 177 -24.72  2 21
    ## 4 4 Cluster_10060  Porites_evermani_scaffold_1977:56240-57240 163 -20.82  2 21
    ## 5 5 Cluster_10060 Porites_evermani_scaffold_209:141486-142486 158 -20.22  2 21
    ## 6 6 Cluster_10060 Porites_evermani_scaffold_285:306283-307283 166 -20.50  2 20
    ##    V7  V8 V9    V10    V11          mRNA     PCC.cor   p_value adjusted_p_value
    ## 1 899 921 17 94.12% 94.12% Peve_00010610 -0.18622118 0.8807518        0.9989356
    ## 2 726 749 20 75.00% 80.00% Peve_00012152          NA        NA               NA
    ## 3 797 816 19 89.47% 89.47% Peve_00000049 -0.03449534 0.9780352        0.9994484
    ## 4 905 927 20 70.00% 85.00% Peve_00013076          NA        NA               NA
    ## 5 781 801 19 68.42% 84.21% Peve_00014408          NA        NA               NA
    ## 6 353 376 20 75.00% 85.00% Peve_00021276          NA        NA               NA

Set function to select genes of interest (ie those that have pvalue \<
0.05)

``` r
topDiffGenes <- function(allScore) {
return(allScore < 0.05)}
```

# 3 FA of all miRNA targets

Functional annotation of all putative miRNA targets

``` r
cor_bind_FA <- left_join(data, annot, by = c("mRNA" = "gene_ID")) %>% distinct()

nrow(cor_bind_FA)
```

    ## [1] 5054

``` r
nrow(cor_bind_FA[!is.na(cor_bind_FA$Gene.Ontology.IDs),])
```

    ## [1] 2106

Of the 5054 putative miRNA targets (5UTR region) predicted by miRanda,
2106 have available annotations

``` r
sig_cor_bind_FA <- cor_bind_FA[cor_bind_FA$p_value < 0.05,]

# Remove rows where all values are NA
sig_cor_bind_FA <- sig_cor_bind_FA[!apply(is.na(sig_cor_bind_FA), 1, all), ]

nrow(sig_cor_bind_FA)
```

    ## [1] 224

``` r
nrow(sig_cor_bind_FA[!is.na(sig_cor_bind_FA$Gene.Ontology.IDs),])
```

    ## [1] 104

Of the 224 putative 5UTR miRNA targets predicted by miRanda that are
also have significantly correlated expression, 104 have available
annotations.

Save

``` r
write.csv(cor_bind_FA, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_targets_FA.csv")
write.csv(sig_cor_bind_FA, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_sig_cor_targets_FA.csv")
```

# 4 FE of specific miRNA’s targets (all targets)

Create topGO function for use with miRNA names

``` r
miRNA_topGO_FE <- function(miRNA.name, input_interactions) {
  
  #Isolate genes in our input module of interest
  interacting_genes <- input_interactions %>%
    filter(miRNA == miRNA.name) %>%
    pull(mRNA)
  
  if (length(interacting_genes) > 0 && any(all_genes %in% interacting_genes)) {
    # Create factor for all reference genes, where 1 represents module membership and 0 means the gene is not in module of interest
    gene_list <- factor(as.integer(all_genes %in% interacting_genes))
    names(gene_list) <- all_genes
    str(gene_list)
    
    ## Biological Process ##
    # Create topGO object
    GO_BP <- new("topGOdata", 
                ontology = "BP", # Biological Process
                allGenes = gene_list,
                annot = annFUN.gene2GO, 
                gene2GO = gene2go_list,
                geneSel=topDiffGenes)
    
    # Run GO enrichment test
    GO_BP_FE <- runTest(GO_BP, algorithm = "weight01", statistic = "fisher")
    # View the results
    GO_BP_results <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
    # Filter by significant results
    GO_BP_results$Fisher<-as.numeric(GO_BP_results$Fisher)
    GO_BP_results_sig<-GO_BP_results[GO_BP_results$Fisher<0.05,]
    
    
    ## Molecular Function ##
    # Create topGO object
    GO_MF <- new("topGOdata", 
                ontology = "MF", # Molecular Function
                allGenes = gene_list,
                annot = annFUN.gene2GO, 
                gene2GO = gene2go_list,
                geneSel=topDiffGenes)
    
    # Run GO enrichment test
    GO_MF_FE <- runTest(GO_MF, algorithm = "weight01", statistic = "fisher")
    # View the results
    GO_MF_results <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
    # Filter by significant results
    GO_MF_results$Fisher<-as.numeric(GO_MF_results$Fisher)
    GO_MF_results_sig<-GO_MF_results[GO_MF_results$Fisher<0.05,]
  
    # Return
    # Add type column only if results exist
    if (nrow(GO_BP_results_sig) > 0) {
      GO_BP_results_sig$type <- "Biological.Process"
    }
    if (nrow(GO_MF_results_sig) > 0) {
      GO_MF_results_sig$type <- "Molecular.Function"
    }
    GO_results <- rbind(GO_BP_results_sig, GO_MF_results_sig)
    print(GO_results)
  }
}

miRNA_topGO_FE("Cluster_10051", cor_bind_FA)
```

Loop through all miRNA and run functional enrichment on the miRNA’s
targets (all predicted targets)

``` r
interacting_miRNAs <- unique(cor_bind_FA$miRNA) %>% na.omit
results_all_targets <- NULL  # initialize empty df

for(miRNA in interacting_miRNAs) {
  
  # Run topGO enrichment function
  miRNA_results <- miRNA_topGO_FE(miRNA, cor_bind_FA)
  
  # Only keep results if not empty
  if (!is.null(miRNA_results) && nrow(miRNA_results) > 0)  {
    
    # Add the miRNA source column
    miRNA_results$miRNA <- miRNA

    # Bind to the accumulating results data frame
    results_all_targets <- rbind(results_all_targets, miRNA_results)
  }
}
```

    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 3 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 3:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (32 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (30 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 15 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   4 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (43 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (94 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (144 eliminated genes)

    ##        GO.ID                                    Term Annotated Significant
    ## 1 GO:0004508 steroid 17-alpha-monooxygenase activity         4           1
    ## 2 GO:0003714      transcription corepressor activity         5           1
    ##   Expected Fisher               type
    ## 1     0.02  0.020 Molecular.Function
    ## 2     0.02  0.024 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 62 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 9:   3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 7:   4 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 6:   10 nodes to be scored   (22 eliminated genes)

    ## 
    ##   Level 5:   14 nodes to be scored   (37 eliminated genes)

    ## 
    ##   Level 4:   10 nodes to be scored   (42 eliminated genes)

    ## 
    ##   Level 3:   6 nodes to be scored    (60 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (111 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (132 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (51 eliminated genes)

    ##        GO.ID                                       Term Annotated Significant
    ## 1 GO:0005355 glucose transmembrane transporter activity         1           1
    ##   Expected Fisher               type
    ## 1        0 0.0016 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 23 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (51 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (78 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (87 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (95 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 36 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   4 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   4 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 5:   7 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 4:   8 nodes to be scored    (68 eliminated genes)

    ## 
    ##   Level 3:   7 nodes to be scored    (119 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (266 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (408 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0000340  RNA 7-methylguanosine cap binding         2           1     0.02
    ## 2 GO:0001664 G protein-coupled receptor binding         3           1     0.02
    ##   Fisher               type
    ## 1  0.016 Molecular.Function
    ## 2  0.024 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (3 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0004658 propionyl-CoA carboxylase activity         1           1        0
    ##   Fisher               type
    ## 1 0.0016 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (23 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (34 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (76 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 39 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   4 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 6:   5 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 5:   6 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 4:   7 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 3:   9 nodes to be scored    (85 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (164 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (182 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 21 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (46 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (51 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (89 eliminated genes)

    ##        GO.ID                                     Term Annotated Significant
    ## 1 GO:0006598              polyamine catabolic process         1           1
    ## 2 GO:0004303 estradiol 17-beta-dehydrogenase activity         3           1
    ## 3 GO:0005267               potassium channel activity         6           1
    ##   Expected Fisher               type
    ## 1     0.01 0.0110 Biological.Process
    ## 2     0.01 0.0098 Molecular.Function
    ## 3     0.02 0.0196 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 48 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  1 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 10:  1 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 9:   2 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 8:   3 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (7 eliminated genes)

    ## 
    ##   Level 6:   6 nodes to be scored    (13 eliminated genes)

    ## 
    ##   Level 5:   7 nodes to be scored    (63 eliminated genes)

    ## 
    ##   Level 4:   10 nodes to be scored   (71 eliminated genes)

    ## 
    ##   Level 3:   7 nodes to be scored    (169 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (184 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (246 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID                              Term Annotated Significant Expected
    ## 1 GO:0001817 regulation of cytokine production         1           1     0.01
    ## 2 GO:0001501       skeletal system development         4           1     0.05
    ## 3 GO:0002098  tRNA wobble uridine modification         4           1     0.05
    ##   Fisher               type
    ## 1  0.011 Biological.Process
    ## 2  0.045 Biological.Process
    ## 3  0.045 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (41 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (42 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (52 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ##        GO.ID                Term Annotated Significant Expected Fisher
    ## 1 GO:0000281 mitotic cytokinesis         6           1     0.02  0.017
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (41 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (46 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (128 eliminated genes)

    ##        GO.ID                             Term Annotated Significant Expected
    ## 1 GO:0004713 protein tyrosine kinase activity        18           1     0.03
    ##   Fisher               type
    ## 1  0.029 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 26 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (28 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (36 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (118 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (151 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (272 eliminated genes)

    ##        GO.ID                                  Term Annotated Significant
    ## 1 GO:0004725 protein tyrosine phosphatase activity         2           1
    ## 2 GO:0005267            potassium channel activity         6           1
    ##   Expected Fisher               type
    ## 1     0.01 0.0098 Molecular.Function
    ## 2     0.03 0.0292 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (69 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (3 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (43 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (49 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (156 eliminated genes)

    ##        GO.ID                                     Term Annotated Significant
    ## 1 GO:0001822                       kidney development         9           1
    ## 2 GO:0004303 estradiol 17-beta-dehydrogenase activity         3           1
    ##   Expected Fisher               type
    ## 1     0.03 0.0250 Biological.Process
    ## 2     0.01 0.0098 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 35 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 11:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 5:   6 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 3:   7 nodes to be scored    (117 eliminated genes)

    ## 
    ##   Level 2:   6 nodes to be scored    (140 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (161 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (14 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (48 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (158 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0000077    DNA damage checkpoint signaling         4           1     0.02
    ## 2 GO:0008061                     chitin binding         8           1     0.03
    ## 3 GO:0004252 serine-type endopeptidase activity        14           1     0.05
    ##   Fisher               type
    ## 1  0.023 Biological.Process
    ## 2  0.026 Molecular.Function
    ## 3  0.045 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 17 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (42 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (53 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (56 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 21 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (28 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (160 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (187 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (304 eliminated genes)

    ##        GO.ID                                 Term Annotated Significant
    ## 1 GO:0000070 mitotic sister chromatid segregation         2           1
    ## 2 GO:0000146         microfilament motor activity         2           1
    ##   Expected Fisher               type
    ## 1     0.01 0.0057 Biological.Process
    ## 2     0.02 0.0160 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 82 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  4 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 9:   4 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 8:   3 nodes to be scored    (53 eliminated genes)

    ## 
    ##   Level 7:   5 nodes to be scored    (54 eliminated genes)

    ## 
    ##   Level 6:   13 nodes to be scored   (55 eliminated genes)

    ## 
    ##   Level 5:   18 nodes to be scored   (74 eliminated genes)

    ## 
    ##   Level 4:   13 nodes to be scored   (84 eliminated genes)

    ## 
    ##   Level 3:   9 nodes to be scored    (127 eliminated genes)

    ## 
    ##   Level 2:   6 nodes to be scored    (187 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (219 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (11 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (12 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (111 eliminated genes)

    ##        GO.ID                             Term Annotated Significant Expected
    ## 1 GO:0001666              response to hypoxia         4           1     0.03
    ## 2 GO:0004364 glutathione transferase activity         1           1     0.00
    ## 3 GO:0003779                    actin binding        10           1     0.03
    ##   Fisher               type
    ## 1 0.0340 Biological.Process
    ## 2 0.0033 Molecular.Function
    ## 3 0.0325 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (30 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (23 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (91 eliminated genes)

    ##        GO.ID                 Term Annotated Significant Expected Fisher
    ## 1 GO:0004518    nuclease activity         8           1     0.03  0.026
    ## 2 GO:0001540 amyloid-beta binding        15           1     0.05  0.049
    ##                 type
    ## 1 Molecular.Function
    ## 2 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 22 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (34 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (68 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (90 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (98 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 32 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 5:   5 nodes to be scored    (37 eliminated genes)

    ## 
    ##   Level 4:   6 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 3:   8 nodes to be scored    (175 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (250 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (393 eliminated genes)

    ##        GO.ID                      Term Annotated Significant Expected Fisher
    ## 1 GO:0006814      sodium ion transport         1           1     0.01 0.0057
    ## 2 GO:0004966 galanin receptor activity         5           1     0.04 0.0400
    ##                 type
    ## 1 Biological.Process
    ## 2 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 40 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 7:   4 nodes to be scored    (17 eliminated genes)

    ## 
    ##   Level 6:   4 nodes to be scored    (17 eliminated genes)

    ## 
    ##   Level 5:   6 nodes to be scored    (19 eliminated genes)

    ## 
    ##   Level 4:   7 nodes to be scored    (26 eliminated genes)

    ## 
    ##   Level 3:   9 nodes to be scored    (74 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (140 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (163 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (16 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (122 eliminated genes)

    ##        GO.ID                                 Term Annotated Significant
    ## 1 GO:0001696               gastric acid secretion         2           1
    ## 2 GO:0000070 mitotic sister chromatid segregation         2           1
    ## 3 GO:0004222        metalloendopeptidase activity        16           1
    ##   Expected Fisher               type
    ## 1     0.02  0.017 Biological.Process
    ## 2     0.02  0.017 Biological.Process
    ## 3     0.03  0.026 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 30 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   4 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 5:   6 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (93 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (136 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (151 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (23 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (111 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0001937 negative regulation of endothelial cell proliferati...         2
    ## 2 GO:0008376               acetylgalactosaminyltransferase activity         2
    ## 3 GO:0005102                             signaling receptor binding        13
    ##   Significant Expected Fisher               type
    ## 1           1     0.01 0.0110 Biological.Process
    ## 2           1     0.01 0.0065 Molecular.Function
    ## 3           1     0.04 0.0421 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 8 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (30 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 30 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (32 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (33 eliminated genes)

    ## 
    ##   Level 6:   4 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 5:   6 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (93 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (136 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (151 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (23 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (111 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0001937 negative regulation of endothelial cell proliferati...         2
    ## 2 GO:0008376               acetylgalactosaminyltransferase activity         2
    ## 3 GO:0005102                             signaling receptor binding        13
    ##   Significant Expected Fisher               type
    ## 1           1     0.01 0.0110 Biological.Process
    ## 2           1     0.01 0.0065 Molecular.Function
    ## 3           1     0.04 0.0421 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 13 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (6 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (39 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (58 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (58 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (129 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0001409 guanine nucleotide transmembrane transporter activi...         6
    ##   Significant Expected Fisher               type
    ## 1           1     0.02   0.02 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 20 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (59 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (111 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (117 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 13 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (5 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (64 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (124 eliminated genes)

    ##        GO.ID                      Term Annotated Significant Expected Fisher
    ## 1 GO:0001054 RNA polymerase I activity         1           1     0.00 0.0033
    ## 2 GO:0004497    monooxygenase activity        13           1     0.04 0.0421
    ##                 type
    ## 1 Molecular.Function
    ## 2 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 6 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 5:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (27 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (71 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (99 eliminated genes)

    ##        GO.ID        Term Annotated Significant Expected Fisher
    ## 1 GO:0003723 RNA binding        27           1     0.04  0.044
    ##                 type
    ## 1 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (87 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (14 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (122 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0001523         retinoid metabolic process         2           1     0.01
    ## 2 GO:0004252 serine-type endopeptidase activity        14           1     0.02
    ##   Fisher               type
    ## 1 0.0057 Biological.Process
    ## 2 0.0230 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (87 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 11 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (16 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (40 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (122 eliminated genes)

    ##        GO.ID                               Term Annotated Significant Expected
    ## 1 GO:0001523         retinoid metabolic process         2           1     0.01
    ## 2 GO:0004177            aminopeptidase activity         2           1     0.01
    ## 3 GO:0004252 serine-type endopeptidase activity        14           1     0.05
    ##   Fisher               type
    ## 1 0.0057 Biological.Process
    ## 2 0.0065 Molecular.Function
    ## 3 0.0453 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (4 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (87 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 12 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 6:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (55 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (92 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (200 eliminated genes)

    ##        GO.ID                       Term Annotated Significant Expected Fisher
    ## 1 GO:0001523 retinoid metabolic process         2           1     0.01 0.0057
    ## 2 GO:0004177    aminopeptidase activity         2           1     0.01 0.0065
    ##                 type
    ## 1 Biological.Process
    ## 2 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 18 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (34 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (42 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (127 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 65 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   4 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 7:   7 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 6:   11 nodes to be scored   (32 eliminated genes)

    ## 
    ##   Level 5:   11 nodes to be scored   (67 eliminated genes)

    ## 
    ##   Level 4:   13 nodes to be scored   (118 eliminated genes)

    ## 
    ##   Level 3:   10 nodes to be scored   (237 eliminated genes)

    ## 
    ##   Level 2:   4 nodes to be scored    (333 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (467 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0001523                             retinoid metabolic process         2
    ## 2 GO:0005248                  voltage-gated sodium channel activity         1
    ## 3 GO:0005109                                       frizzled binding         1
    ## 4 GO:0003729                                           mRNA binding         2
    ## 5 GO:0001601                           peptide YY receptor activity         2
    ## 6 GO:0003831 beta-N-acetylglucosaminylglycopeptide beta-1,4-gala...         2
    ## 7 GO:0004303               estradiol 17-beta-dehydrogenase activity         3
    ##   Significant Expected Fisher               type
    ## 1           1     0.01  0.011 Biological.Process
    ## 2           1     0.02  0.016 Molecular.Function
    ## 3           1     0.02  0.016 Molecular.Function
    ## 4           1     0.03  0.032 Molecular.Function
    ## 5           1     0.03  0.032 Molecular.Function
    ## 6           1     0.03  0.032 Molecular.Function
    ## 7           1     0.05  0.048 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)

``` r
head(results_all_targets)
```

    ##        GO.ID                                       Term Annotated Significant
    ## 1 GO:0004508    steroid 17-alpha-monooxygenase activity         4           1
    ## 2 GO:0003714         transcription corepressor activity         5           1
    ## 3 GO:0005355 glucose transmembrane transporter activity         1           1
    ## 4 GO:0000340          RNA 7-methylguanosine cap binding         2           1
    ## 5 GO:0001664         G protein-coupled receptor binding         3           1
    ## 6 GO:0004658         propionyl-CoA carboxylase activity         1           1
    ##   Expected Fisher               type         miRNA
    ## 1     0.02 0.0200 Molecular.Function Cluster_11134
    ## 2     0.02 0.0240 Molecular.Function Cluster_11134
    ## 3     0.00 0.0016 Molecular.Function Cluster_11135
    ## 4     0.02 0.0160 Molecular.Function  Cluster_1167
    ## 5     0.02 0.0240 Molecular.Function  Cluster_1167
    ## 6     0.00 0.0016 Molecular.Function Cluster_14500

Save results

``` r
write.csv(results_all_targets, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_all_targets_topGO_FE.csv")
```

# 5 FE of specific miRNA’s targets (significant cor targets)

Loop through all miRNA and run functional enrichment on the miRNA’s
significantly correlated targets

``` r
interacting_miRNAs_sig <- unique(sig_cor_bind_FA$miRNA) %>% na.omit
results_sig_cor_targets <- NULL  # initialize empty df

for(miRNA in interacting_miRNAs_sig) {
  
  # Run topGO enrichment function
  miRNA_results <- miRNA_topGO_FE(miRNA, sig_cor_bind_FA)
  
  # Only keep results if not empty
  if (!is.null(miRNA_results) && nrow(miRNA_results) > 0) {
    
    # Add the miRNA source column
    miRNA_results$miRNA <- miRNA

    # Bind to the accumulating results data frame
    results_sig_cor_targets <- rbind(results_sig_cor_targets, miRNA_results)
  }
}
```

    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 24 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (65 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (79 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (172 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (278 eliminated genes)

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0004714 transmembrane receptor protein tyrosine kinase acti...         8
    ##   Significant Expected Fisher               type
    ## 1           1     0.03  0.026 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (14 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (14 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (17 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID                                            Term Annotated
    ## 1 GO:0002175 protein localization to paranode region of axon        14
    ##   Significant Expected Fisher               type
    ## 1           1     0.04   0.04 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   2 nodes to be scored    (18 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (35 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (41 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (46 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (128 eliminated genes)

    ##        GO.ID                             Term Annotated Significant Expected
    ## 1 GO:0004713 protein tyrosine kinase activity        18           1     0.03
    ##   Fisher               type
    ## 1  0.029 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 10 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 4:   2 nodes to be scored    (9 eliminated genes)

    ## 
    ##   Level 3:   1 nodes to be scored    (62 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (69 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (78 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID               Term Annotated Significant Expected Fisher
    ## 1 GO:0001822 kidney development         9           1     0.03  0.025
    ##                 type
    ## 1 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 4 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 3:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (2 eliminated genes)

    ##        GO.ID                         Term Annotated Significant Expected Fisher
    ## 1 GO:0000146 microfilament motor activity         2           1        0 0.0033
    ##                 type
    ## 1 Molecular.Function
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 17 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   2 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 5:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 4:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 3:   3 nodes to be scored    (42 eliminated genes)

    ## 
    ##   Level 2:   1 nodes to be scored    (53 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (56 eliminated genes)

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ##        GO.ID                                 Term Annotated Significant
    ## 1 GO:0000070 mitotic sister chromatid segregation         2           1
    ##   Expected Fisher               type
    ## 1     0.01 0.0057 Biological.Process
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)
    ##  Factor w/ 2 levels "0","1": 1 1 1 1 1 1 1 1 1 1 ...
    ##  - attr(*, "names")= chr [1:31501] "STRG.24381" "STRG.24380" "STRG.15042" "STRG.8662" ...

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

    ## [1] GO.ID       Term        Annotated   Significant Expected    Fisher     
    ## <0 rows> (or 0-length row.names)

``` r
head(results_sig_cor_targets)
```

    ##        GO.ID                                                   Term Annotated
    ## 1 GO:0004714 transmembrane receptor protein tyrosine kinase acti...         8
    ## 2 GO:0002175        protein localization to paranode region of axon        14
    ## 3 GO:0004713                       protein tyrosine kinase activity        18
    ## 4 GO:0001822                                     kidney development         9
    ## 5 GO:0000146                           microfilament motor activity         2
    ## 6 GO:0000070                   mitotic sister chromatid segregation         2
    ##   Significant Expected Fisher               type         miRNA
    ## 1           1     0.03 0.0260 Molecular.Function  Cluster_1167
    ## 2           1     0.04 0.0400 Biological.Process Cluster_15726
    ## 3           1     0.03 0.0290 Molecular.Function  Cluster_2854
    ## 4           1     0.03 0.0250 Biological.Process  Cluster_4079
    ## 5           1     0.00 0.0033 Molecular.Function  Cluster_4629
    ## 6           1     0.01 0.0057 Biological.Process   Cluster_589

Save results

``` r
write.csv(results_sig_cor_targets, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_sig_cor_targets_topGO_FE.csv")
```

# 6 FE of all targets negatively correlated with miRNAs (regardless of correlation significance)

Filter PCC miranda data

``` r
# Filter so that only negative correlations remain 
neg_corr_data <- data %>%
  filter(PCC.cor < 0)
```

Make list of target genes for input to topGO

``` r
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(neg_corr_data$mRNA))
```

``` r
# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs, regardless of
correlation significance.

### 6.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 171 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  5 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 10:  6 nodes to be scored    (25 eliminated genes)

    ## 
    ##   Level 9:   11 nodes to be scored   (29 eliminated genes)

    ## 
    ##   Level 8:   13 nodes to be scored   (62 eliminated genes)

    ## 
    ##   Level 7:   19 nodes to be scored   (85 eliminated genes)

    ## 
    ##   Level 6:   28 nodes to be scored   (102 eliminated genes)

    ## 
    ##   Level 5:   32 nodes to be scored   (148 eliminated genes)

    ## 
    ##   Level 4:   24 nodes to be scored   (185 eliminated genes)

    ## 
    ##   Level 3:   19 nodes to be scored   (266 eliminated genes)

    ## 
    ##   Level 2:   8 nodes to be scored    (296 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (324 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
neg_cor_gene2go$GO.ID <- trimws(neg_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- neg_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- GO_BP_En_sig_gene %>% distinct()
```

### 6.0.2 Cellular Components

### 6.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 96 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 10:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 9:   4 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   6 nodes to be scored    (1 eliminated genes)

    ## 
    ##   Level 7:   9 nodes to be scored    (26 eliminated genes)

    ## 
    ##   Level 6:   16 nodes to be scored   (35 eliminated genes)

    ## 
    ##   Level 5:   19 nodes to be scored   (72 eliminated genes)

    ## 
    ##   Level 4:   21 nodes to be scored   (157 eliminated genes)

    ## 
    ##   Level 3:   14 nodes to be scored   (334 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (448 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (551 eliminated genes)

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
neg_cor_gene2go$GO.ID <- trimws(neg_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- neg_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 6.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_neg_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_neg_corr_df`

``` r
test <- GO_neg_corr_df %>%
  inner_join(neg_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_neg_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-29-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_neg_corr_target_enrichment.pdf", plot, width = 20, height = 35, dpi = 300)
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_neg_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "estradiol 17-beta-dehydrogenase activity"
    ## [2] "voltage-gated sodium channel activity"   
    ## [3] "frizzled binding"                        
    ## [4] "glutathione transferase activity"        
    ## [5] "RNA polymerase I activity"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-30-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_top10GO_neg_corr_target_enrichment.pdf", plot, width = 20, height = 25, dpi = 300)
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_top10GO_neg_corr_target_enrichment.png", plot, width = 20, height = 25, dpi = 300)
```

# 7 FE of all targets positively correlated with miRNAs (regardless of correlation significance)

Filter PCC miranda data

``` r
# Filter so that only positive correlations remain 
pos_corr_data <- data %>%
  filter(PCC.cor > 0)

# Make list of target genes for input to topGO
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(pos_corr_data$mRNA))

# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs with positively
correlated expression, regardless of correlation significance.

### 7.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 138 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 13:  1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 12:  2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 11:  3 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 10:  5 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 9:   6 nodes to be scored    (21 eliminated genes)

    ## 
    ##   Level 8:   6 nodes to be scored    (59 eliminated genes)

    ## 
    ##   Level 7:   13 nodes to be scored   (71 eliminated genes)

    ## 
    ##   Level 6:   21 nodes to be scored   (74 eliminated genes)

    ## 
    ##   Level 5:   29 nodes to be scored   (138 eliminated genes)

    ## 
    ##   Level 4:   22 nodes to be scored   (158 eliminated genes)

    ## 
    ##   Level 3:   21 nodes to be scored   (260 eliminated genes)

    ## 
    ##   Level 2:   8 nodes to be scored    (289 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (335 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
pos_cor_gene2go$GO.ID <- trimws(pos_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- pos_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- unique(GO_BP_En_sig_gene)
```

### 7.0.2 Cellular Components

### 7.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 87 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   4 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   9 nodes to be scored    (22 eliminated genes)

    ## 
    ##   Level 6:   14 nodes to be scored   (36 eliminated genes)

    ## 
    ##   Level 5:   18 nodes to be scored   (86 eliminated genes)

    ## 
    ##   Level 4:   19 nodes to be scored   (173 eliminated genes)

    ## 
    ##   Level 3:   15 nodes to be scored   (366 eliminated genes)

    ## 
    ##   Level 2:   6 nodes to be scored    (445 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (563 eliminated genes)

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
pos_cor_gene2go$GO.ID <- trimws(pos_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- pos_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 7.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_pos_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_pos_corr_df`

``` r
test <- GO_pos_corr_df %>%
  inner_join(pos_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_pos_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-44-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_pos_corr_target_enrichment.pdf", plot, width = 20, height = 35, dpi = 300)
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_pos_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "glucose transmembrane transporter activity"
    ## [2] "propionyl-CoA carboxylase activity"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-45-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_top10GO_pos_corr_target_enrichment.pdf", plot, width = 20, height = 25, dpi = 300)
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_top10GO_pos_corr_target_enrichment.png", plot, width = 20, height = 25, dpi = 300)
```

# 8 FE of all targets significantly negatively correlated with miRNAs

Filter PCC miranda data

``` r
# Filter so that only significant negative correlations remain 
sig_neg_corr_data <- data %>%
  filter(PCC.cor < 0) %>%
  filter(p_value < 0.05)

# Make list of target genes for input to topGO
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(sig_neg_corr_data$mRNA))

# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs with significantly
negatively correlated expression.

### 8.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 26 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 9:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 8:   2 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (2 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (11 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (11 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (11 eliminated genes)

    ## 
    ##   Level 3:   4 nodes to be scored    (104 eliminated genes)

    ## 
    ##   Level 2:   3 nodes to be scored    (122 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (134 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_neg_cor_gene2go$GO.ID <- trimws(sig_neg_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- sig_neg_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- unique(GO_BP_En_sig_gene)
```

### 8.0.2 Cellular Components

### 8.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 0 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## Warning in getSigGroups(object, test.stat): No enrichment can pe performed -
    ## there are no feasible GO terms!

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_neg_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_neg_cor_gene2go$GO.ID <- trimws(sig_neg_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- sig_neg_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 8.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_sig_neg_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_sig_neg_corr_df`

``` r
test <- GO_sig_neg_corr_df %>%
  inner_join(sig_neg_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_sig_neg_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-59-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_sig_neg_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "mitotic sister chromatid segregation"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-60-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_top10GO_sig_neg_corr_target_enrichment.pdf", plot, width = 20, height = 25, dpi = 300)
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_top10GO_sig_neg_corr_target_enrichment.png", plot, width = 20, height = 25, dpi = 300)
```

# 9 FE of all targets significantly positively correlated with miRNAs

Filter PCC miranda data

``` r
# Filter so that only significant negative correlations remain 
sig_pos_corr_data <- data %>%
  filter(PCC.cor > 0) %>%
  filter(p_value < 0.05)

# Make list of target genes for input to topGO
# Genes of interest - ie those targeted by miRNAs
target_genes <- as.character(unique(sig_pos_corr_data$mRNA))

# Apply 1 or 0 if gene is gene of interest 
GeneList <- factor(as.integer(all_genes %in% target_genes))
names(GeneList) <- all_genes
```

The following code will perform GO enrichment using the weighted
Fisher’s exact test to assess whether specific GO terms are
overrepresented in the genes targeted by miRNAs with significantly
positively correlated expression.

### 9.0.1 Biological Processes

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_BP <-new("topGOdata", 
            ontology="BP", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 115 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 747 GO terms and 1413 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 353 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_BP_FE <- runTest(GO_BP, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 9 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 7:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 5:   1 nodes to be scored    (14 eliminated genes)

    ## 
    ##   Level 4:   1 nodes to be scored    (14 eliminated genes)

    ## 
    ##   Level 3:   2 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 2:   2 nodes to be scored    (15 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (17 eliminated genes)

Generate results table

``` r
GO_BP_En <- GenTable(GO_BP, Fisher = GO_BP_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_BP_En$Fisher<-as.numeric(GO_BP_En$Fisher)
GO_BP_En_sig<-GO_BP_En[GO_BP_En$Fisher<0.05,]
```

Merge `GO_BP_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_pos_cor_gene2go$GO.ID <- trimws(sig_pos_cor_gene2go$GO.ID)
GO_BP_En_sig$GO.ID <- trimws(GO_BP_En_sig$GO.ID)

# Join the datasets based on GO term
GO_BP_En_sig_gene <- sig_pos_cor_gene2go %>%
  left_join(GO_BP_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_BP_En_sig_gene$ontology <- "Biological Processes"

# Keep only unique rows 
GO_BP_En_sig_gene <- unique(GO_BP_En_sig_gene)
```

### 9.0.2 Cellular Components

### 9.0.3 Molecular Functions

Create `topGOdata` object, which is required for topGO analysis

``` r
GO_MF <-new("topGOdata", 
            ontology="MF", 
            gene2GO=gene2go_list, 
            allGenes=GeneList, 
            annot = annFUN.gene2GO, 
            geneSel=topDiffGenes)
```

    ## 
    ## Building most specific GOs .....

    ##  ( 183 GO terms found. )

    ## 
    ## Build GO DAG topology ..........

    ##  ( 467 GO terms and 589 relations. )

    ## 
    ## Annotating nodes ...............

    ##  ( 611 genes annotated to the GO terms. )

Run GO enrichment test

``` r
GO_MF_FE <- runTest(GO_MF, algorithm="weight01", statistic="fisher")
```

    ## 
    ##           -- Weight01 Algorithm -- 
    ## 
    ##       the algorithm is scoring 27 nontrivial nodes
    ##       parameters: 
    ##           test statistic: fisher

    ## 
    ##   Level 8:   1 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 7:   3 nodes to be scored    (0 eliminated genes)

    ## 
    ##   Level 6:   3 nodes to be scored    (8 eliminated genes)

    ## 
    ##   Level 5:   4 nodes to be scored    (38 eliminated genes)

    ## 
    ##   Level 4:   5 nodes to be scored    (65 eliminated genes)

    ## 
    ##   Level 3:   5 nodes to be scored    (79 eliminated genes)

    ## 
    ##   Level 2:   5 nodes to be scored    (172 eliminated genes)

    ## 
    ##   Level 1:   1 nodes to be scored    (280 eliminated genes)

Generate results table

``` r
GO_MF_En <- GenTable(GO_MF, Fisher = GO_MF_FE, orderBy = "Fisher",  topNodes = 100, numChar = 51)
```

Only taking the top 100 GO terms

Filter by significant results

``` r
GO_MF_En$Fisher<-as.numeric(GO_MF_En$Fisher)
GO_MF_En_sig<-GO_MF_En[GO_MF_En$Fisher<0.05,]
```

Merge `GO_MF_En_sig` with GO and gene info.

``` r
# Separate GO terms 
sig_pos_cor_gene2go <- gene2go %>%
  separate_rows(GO.ID, sep = ";")

# Ensure GO terms in both datasets are formatted similarly (trim whitespaces)
sig_pos_cor_gene2go$GO.ID <- trimws(sig_pos_cor_gene2go$GO.ID)
GO_MF_En_sig$GO.ID <- trimws(GO_MF_En_sig$GO.ID)

# Join the datasets based on GO term
GO_MF_En_sig_gene <- sig_pos_cor_gene2go %>%
  left_join(GO_MF_En_sig, by = "GO.ID") %>%
  na.omit()

# Add ontology column 
GO_MF_En_sig_gene$ontology <- "Molecular Functions"

# Keep only unique rows 
GO_MF_En_sig_gene <- unique(GO_MF_En_sig_gene)
```

### 9.0.4 Join ontologies

Bind so there is a df that has significantly enriched GO terms for all
ontologies

``` r
GO_sig_pos_corr_df <- rbind(GO_BP_En_sig_gene, GO_MF_En_sig_gene)
```

Merge with `GO_pos_corr_df`

``` r
test <- GO_sig_pos_corr_df %>%
  inner_join(sig_pos_corr_data, by = c("gene_ID" = "mRNA")) #%>%
  #mutate(direction = ifelse(PCC.cor > 0, "Positive", "Negative")) #%>%
  #filter(ontology != "Cellular Components") #%>%
  #filter(p_value < 0.1)

# Save as csv 
write.csv(test, "../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_sig_pos_corr_target_enrichment.csv")
```

Plot!

``` r
plot<-ggplot(test, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(shape = 21, size = 5) + 
  #scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 24) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-74-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_sig_pos_corr_target_enrichment.pdf", plot, width = 20, height = 35, dpi = 300)
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/miRNA_5UTR_topGO_sig_pos_corr_target_enrichment.png", plot, width = 20, height = 35, dpi = 300)
```

Examine the top 10 most significant GO terms for BP and MF

``` r
# Function to get top 5 unique terms
get_top_10_unique <- function(data, ontology_type) {
  data %>%
    filter(ontology == ontology_type) %>%
    arrange(Fisher) %>%
    distinct(Term, .keep_all = TRUE) %>%
    slice_head(n = 10)
}

# Get top 5 unique Biological Processes
top_10_BP <- get_top_10_unique(test, "Biological Processes")

# Get top 5 unique Molecular Functions
top_10_MF <- get_top_10_unique(test, "Molecular Functions")

# Combine results
top_10_combined <- bind_rows(top_10_BP, top_10_MF)
unique(top_10_combined$Term)
```

    ## [1] "protein localization to paranode region of axon"
    ## [2] "microfilament motor activity"                   
    ## [3] "protein tyrosine kinase activity"

``` r
# Plot 
top_10_combined <- top_10_combined %>%
  arrange(desc(Fisher)) %>%
  mutate(Term = factor(Term, levels = unique(Term)))

plot<-ggplot(top_10_combined, aes(x = Term, y = Fisher, fill = p_value)) +
  #ylim(0, 1) +
  #geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+
  #geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 
  geom_point(size = 10, color = "black") + 
  scale_size(range = c(2, 20)) + 
  xlab('') + 
  ylab("Fisher p-value") +
  theme_bw(base_size = 35) +
  facet_grid(vars(ontology), scales = "free", space = "free_y") +
  coord_flip(); plot
```

![](10.12-Peve-mRNA-miRNA-interactions-FE-5UTR_files/figure-gfm/unnamed-chunk-75-1.png)<!-- -->

``` r
# Save plot 
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/top10GO_sig_pos_corr_target_enrichment.pdf", last_plot(), width = 20, height = 25, dpi = 300)
ggsave("../output/10.12-Peve-mRNA-miRNA-interactions-FE-5UTR/top10GO_sig_pos_corr_target_enrichment.png", last_plot(), width = 20, height = 25, dpi = 300)
```
