---
title: "08-E5-Deep-Dive-Expression-WGBS"
author: "Zoe Dellaert"
date: "2024-10-01"
always_allow_html: true
output: 
  github_document:
    toc: true
    toc_depth: 3
    number_sections: true
    html_preview: true 
---

## E5 Deep Dive Expression WGBS

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = TRUE,        # Evaluate code chunks
  warning = TRUE,     # Hide warnings
  message = TRUE,     # Hide messages
  comment = "" ,        # Prevents appending '##' to beginning of lines in code output
  quiet = FALSE
)
```

# Create a Bash variables file

This allows usage of Bash variables (e.g. paths to common directories) across R Markdown chunks.
```{bash bash-vars}
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export repo_dir=/work/pi_hputnam_uri_edu/deep-dive-expression'
echo 'export code_dir=${repo_dir}/D-Apul/code'
echo 'export genome_dir=${repo_dir}/D-Apul/data/genome'
echo 'export output_dir_top=${repo_dir}/D-Apul/output/08-Apul-WGBS'
echo 'export sim_data_dir=${repo_dir}/D-Apul/data/08-Apul-WGBS/sim-data'
echo ""

echo "# Paths to programs"
echo 'export sherman=/work/pi_hputnam_uri_edu/pgrams/Sherman/Sherman'
echo ""

echo "# Programs associative array"
echo "declare -A programs_array"
echo "programs_array=("
echo '[sherman]="${sherman}"'
echo ")"
} > .bashvars

cat .bashvars
```

## Genome

```{bash, eval=FALSE}
cd ../data/genome

wget -O Apulcra-genome.fa "https://osf.io/download/kn96u/"
```

## simulate WGBS data for *Acropora pulchra* on Unity

#### Install Sherman on unity in shared directory and make executable

```{bash install-sherman, eval=FALSE}
cd /work/pi_hputnam_uri_edu/pgrams

git clone https://github.com/FelixKrueger/Sherman.git

cd Sherman
chmod +x Sherman
```

```{bash run-sherman, eval=FALSE}
source .bashvars

${programs_array[sherman]} \
--genome ${genome_dir} \
--length 150 \
--number_of_seqs 1000000 \
--paired_end \
--output_dir ${sim_data_dir}

gzip ${sim_data_dir}/*.fastq
```

### Run nfcore-methylseq based on [Danielle's Notebook](https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2023-03-13-P.meandrina-WGBS-Workflow-Host.md)

Create input csv file containing the simulated data 

```{bash}
source .bashvars

echo "sample,fastq_1,fastq_2" > ${sim_data_dir}/input.csv
echo "SIM_1,${sim_data_dir}/simulated_1.fastq.gz,${sim_data_dir}/simulated_2.fastq.gz" >> ${sim_data_dir}/input.csv

# Optional: Display the file content to verify
cat ${sim_data_dir}/input.csv
```

To run on unity, I also needed to set up a sigularity cache directory

```{bash, eval=FALSE}
cd /work/pi_hputnam_uri_edu/
mkdir .apptainer
mkdir .apptainer/cache
singularity --version

nano ~/.bash_profile
# add the following text (without the #) to the bottom of the file
# export NXF_SINGULARITY_CACHEDIR=/work/pi_hputnam_uri_edu/.apptainer/cache
```

```{bash}
# Create the job script
cat << 'EOF' > methylseq.sh
#!/usr/bin/env bash
#SBATCH --export=NONE
#SBATCH --nodes=1 --ntasks-per-node=24
#SBATCH --signal=2
#SBATCH --no-requeue
#SBATCH --mem=500GB
#SBATCH --error="%x_error.%j"
#SBATCH --output="%x_output.%j"
#SBATCH -t 24:00:00
#SBATCH --mail-type=BEGIN,END,FAIL

# Load modules
module load nextflow/24.04.2

# Source the bashvars file
source .bashvars

# Navigate to the output directory
cd ${output_dir_top}

# Print variables from bashvars
echo "Simulated data path: ${sim_data_dir}"

export NXF_SINGULARITY_CACHEDIR=/work/pi_hputnam_uri_edu/.apptainer/cache

# Run nextflow methylseq

nextflow run nf-core/methylseq -profile singularity \
--aligner bismark \
--fasta ${genome_dir}/Apulcra-genome.fa \
--save_reference \
--input ${sim_data_dir}/input.csv \
--zymo \
--cytosine_report \
--relax_mismatches \
--unmapped \
--outdir ${output_dir_top}/sim_Apul_methylseq
#--igenomes_ignore \
#-resume

EOF

chmod +x methylseq.sh

# Submit the job to the cluster , go to terminal to do this - it gets confusing in Rmd...
# sbatch methylseq.sh
```

This is running as of 10/2/24.

```{bash, eval=FALSE}
source .bashvars

tail ${output_dir_top}/.nextflow.log
```

```
Oct-02 03:29:18.466 [TaskFinalizer-1] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'PublishDir' minSize=10; maxSize=72; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false
Oct-02 03:29:21.720 [Actor Thread 25] DEBUG nextflow.container.SingularityCache - Singularity pull complete image=https://depot.galaxyproject.org/singularity/trim-galore:0.6.7--hdfd78af_0 path=/work/pi_hputnam_uri_edu/deep-dive-expression/D-Apul/output/08-Apul-WGBS/work/singularity/depot.galaxyproject.org-singularity-trim-galore-0.6.7--hdfd78af_0.img
Oct-02 03:29:21.770 [Task submitter] DEBUG n.executor.local.LocalTaskHandler - Launch cmd line: /bin/bash -ue .command.run
Oct-02 03:29:21.770 [Task submitter] INFO  nextflow.Session - [da/848bb6] Submitted process > NFCORE_METHYLSEQ:METHYLSEQ:TRIMGALORE (SIM)
Oct-02 03:29:50.952 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 3; name: NFCORE_METHYLSEQ:METHYLSEQ:TRIMGALORE (SIM); status: COMPLETED; exit: 0; error: -; workDir: /work/pi_hputnam_uri_edu/deep-dive-expression/D-Apul/output/08-Apul-WGBS/work/da/848bb60dd00624def2ad6805af7d38]
Oct-02 03:29:51.047 [TaskFinalizer-2] DEBUG nextflow.processor.TaskProcessor - Process NFCORE_METHYLSEQ:METHYLSEQ:TRIMGALORE > Skipping output binding because one or more optional files are missing: fileoutparam<2:1>
Oct-02 03:33:47.947 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor local > tasks to be completed: 1 -- submitted tasks are shown below
~> TaskHandler[id: 1; name: NFCORE_METHYLSEQ:METHYLSEQ:PREPARE_GENOME:BISMARK_GENOMEPREPARATION (BismarkIndex/Apulcra-genome.fa); status: RUNNING; exit: -; error: -; workDir: /work/pi_hputnam_uri_edu/deep-dive-expression/D-Apul/output/08-Apul-WGBS/work/a7/e052ad239af94503a7a29185f3481e]
Oct-02 03:38:47.981 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor local > tasks to be completed: 1 -- submitted tasks are shown below
~> TaskHandler[id: 1; name: NFCORE_METHYLSEQ:METHYLSEQ:PREPARE_GENOME:BISMARK_GENOMEPREPARATION (BismarkIndex/Apulcra-genome.fa); status: RUNNING; exit: -; error: -; workDir: /work/pi_hputnam_uri_edu/deep-dive-expression/D-Apul/output/08-Apul-WGBS/work/a7/e052ad239af94503a7a29185f3481e]
```

I had issues with some conflicts when BISMARK tried to index the genome, it wasnt finding the genome file even though it found it to align the data...

So, I looked at [Emma's code](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/_posts/2021-10-21-KBay-Bleaching-Pairs-WGBS-Analysis-Pipeline.md#Test) and we could potentially add the flag --igenomes_ignore to the nfcore script and attemped to rerun the pipeline.


But first I reran all of it with a specified apptainer cache ( /work/pi_hputnam_uri_edu/.apptainer/cache)