---
title: "20-Apul-Orthofinder"
author: "Zoe Dellaert"
date: "2024-12-20"
always_allow_html: true
output: 
  github_document:
    toc: true
    toc_depth: 3
    number_sections: true
    html_preview: true 
---

## E5 Deep Dive Expression Orthofinder for all 3 genomes

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = TRUE,        # Evaluate code chunks
  warning = TRUE,     # Hide warnings
  message = TRUE,     # Hide messages
  comment = "" ,        # Prevents appending '##' to beginning of lines in code output
  quiet = FALSE
)
```

For orthofinder, we need a protein fasta for each genome. I found each of these on our wiki, https://github.com/urol-e5/deep-dive-expression/wiki/00%E2%80%90Genomic-Resources.

- `Apulchra-genome.pep.faa` - https://raw.githubusercontent.com/urol-e5/deep-dive-expression/main/D-Apul/data/Apulchra-genome.pep.faa
- `Porites_evermanni_v1.annot.pep.fa`: https://gannet.fish.washington.edu/seashell/snaps/Porites_evermanni_v1.annot.pep.fa 
- `Pocillopora_meandrina_HIv1.genes.pep.faa` - https://gannet.fish.washington.edu/seashell/snaps/Pocillopora_meandrina_HIv1.genes.pep.faa

```{bash, eval = FALSE}
cd ../data
mkdir -p 20-Apul-Orthofinder/proteins
cd 20-Apul-Orthofinder/proteins

wget https://raw.githubusercontent.com/urol-e5/deep-dive-expression/main/D-Apul/data/Apulchra-genome.pep.faa
wget https://gannet.fish.washington.edu/seashell/snaps/Porites_evermanni_v1.annot.pep.fa
wget https://gannet.fish.washington.edu/seashell/snaps/Pocillopora_meandrina_HIv1.genes.pep.faa
```

Install Orthofinder on unity

```{bash, eval = FALSE}
cd /work/pi_hputnam_uri_edu/pgrams

wget https://github.com/davidemms/OrthoFinder/releases/latest/download/OrthoFinder.tar.gz

tar xzvf OrthoFinder.tar.gz
cd OrthoFinder/

#print help 
./orthofinder -h
```


```{bash, eval = FALSE}
mkdir scripts
nano scripts/orthofinder.sh
```

```{bash, eval = FALSE}
#!/usr/bin/env bash
#SBATCH --export=NONE
#SBATCH --nodes=1 --ntasks-per-node=24
#SBATCH --signal=2
#SBATCH --no-requeue
#SBATCH --mem=200GB
#SBATCH --error=scripts/"%x_error.%j"
#SBATCH --output=scripts/"%x_output.%j"
#SBATCH -t 48:00:00
#SBATCH --mail-type=BEGIN,END,FAIL

# run orthofinder with 20 threads
/work/pi_hputnam_uri_edu/pgrams/OrthoFinder/orthofinder -f ../data/20-Apul-Orthofinder/proteins/ -t 20 -o ../output/20-Apul-Orthofinder/
```

View output summary:

```{bash, eval=FALSE}
cd ../output/20-Apul-Orthofinder/Results_Dec20
cat Comparative_Genomics_Statistics/Statistics_Overall.tsv
```

```
Number of species	3
Number of genes	112750
Number of genes in orthogroups	97235
Number of unassigned genes	15515
Percentage of genes in orthogroups	86.2
Percentage of unassigned genes	13.8
Number of orthogroups	23124
Number of species-specific orthogroups	5160
Number of genes in species-specific orthogroups	24458
Percentage of genes in species-specific orthogroups	21.7
Mean orthogroup size	4.2
Median orthogroup size	3.0
G50 (assigned genes)	4
G50 (all genes)	4
O50 (assigned genes)	5985
O50 (all genes)	7924
Number of orthogroups with all species present	12864
Number of single-copy orthogroups	8082
Date	2024-12-20
Orthogroups file	Orthogroups.tsv
Unassigned genes file	Orthogroups_UnassignedGenes.tsv
Per-species statistics	Statistics_PerSpecies.tsv
Overall statistics	Statistics_Overall.tsv
Orthogroups shared between species	Orthogroups_SpeciesOverlaps.tsv

Average number of genes per-species in orthogroup	Number of orthogroups	Percentage of orthogroups	Number of genes	Percentage of genes
<1	4980	21.5	9960	10.2
'1	14966	64.7	51441	52.9
'2	1709	7.4	11512	11.8
'3	615	2.7	6060	6.2
'4	318	1.4	4081	4.2
'5	171	0.7	2709	2.8
'6	85	0.4	1613	1.7
'7	58	0.3	1279	1.3
'8	44	0.2	1099	1.1
'9	37	0.2	1035	1.1
'10	35	0.2	1084	1.1
11-15	67	0.3	2554	2.6
16-20	18	0.1	999	1.0
21-50	21	0.1	1809	1.9
51-100	0	0.0	0	0.0
101-150	0	0.0	0	0.0
151-200	0	0.0	0	0.0
201-500	0	0.0	0	0.0
501-1000	0	0.0	0	0.0
'1001+	0	0.0	0	0.0

Number of species in orthogroup	Number of orthogroups
1	5160
2	5100
3	12864
```

```{bash, eval = FALSE}
cat Species_Tree/SpeciesTree_rooted.txt
```

```(Porites_evermanni_v1.annot.pep:0.5,(Apulchra-genome.pep:1,Pocillopora_meandrina_HIv1.genes.pep:1)1:0.5);```

Put this info^^ into the [ETE Toolkit tree viewer](http://etetoolkit.org/treeview/) in web browser

![OrthofinderTree](../output/20-Apul-Orthofinder/Results_Dec20/Orthofinder-tree.png)
Analyze data:
