---
title: "20-Apul-Orthofinder"
author: "Zoe Dellaert"
date: "2024-12-20"
always_allow_html: true
output: 
  github_document:
    toc: true
    toc_depth: 3
    number_sections: true
    html_preview: true 
---

## E5 Deep Dive Expression Orthofinder for all 3 genomes

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = TRUE,        # Evaluate code chunks
  warning = TRUE,     # Hide warnings
  message = TRUE,     # Hide messages
  comment = "" ,        # Prevents appending '##' to beginning of lines in code output
  quiet = FALSE
)
```

For orthofinder, we need a protein fasta for each genome. I found each of these on our wiki, https://github.com/urol-e5/deep-dive-expression/wiki/00%E2%80%90Genomic-Resources.

- `Apulchra-genome.pep.faa` - https://raw.githubusercontent.com/urol-e5/deep-dive-expression/main/D-Apul/data/Apulchra-genome.pep.faa
- `Porites_evermanni_v1.annot.pep.fa`: https://gannet.fish.washington.edu/seashell/snaps/Porites_evermanni_v1.annot.pep.fa 
- `Pocillopora_meandrina_HIv1.genes.pep.faa` - https://gannet.fish.washington.edu/seashell/snaps/Pocillopora_meandrina_HIv1.genes.pep.faa

```{bash, eval = FALSE}
cd ../data
mkdir -p 20-Apul-Orthofinder/proteins
cd 20-Apul-Orthofinder/proteins

wget https://raw.githubusercontent.com/urol-e5/deep-dive-expression/main/D-Apul/data/Apulchra-genome.pep.faa
wget https://gannet.fish.washington.edu/seashell/snaps/Porites_evermanni_v1.annot.pep.fa
wget https://gannet.fish.washington.edu/seashell/snaps/Pocillopora_meandrina_HIv1.genes.pep.faa
```

Install Orthofinder on unity

```{bash, eval = FALSE}
cd /work/pi_hputnam_uri_edu/pgrams

wget https://github.com/davidemms/OrthoFinder/releases/latest/download/OrthoFinder.tar.gz

tar xzvf OrthoFinder.tar.gz
cd OrthoFinder/

#print help 
./orthofinder -h
```


```{bash, eval = FALSE}
mkdir scripts
nano scripts/orthofinder.sh
```

```{bash, eval = FALSE}
#!/usr/bin/env bash
#SBATCH --export=NONE
#SBATCH --nodes=1 --ntasks-per-node=24
#SBATCH --signal=2
#SBATCH --no-requeue
#SBATCH --mem=200GB
#SBATCH --error=scripts/"%x_error.%j"
#SBATCH --output=scripts/"%x_output.%j"
#SBATCH -t 48:00:00
#SBATCH --mail-type=BEGIN,END,FAIL

# run orthofinder with 20 threads
/work/pi_hputnam_uri_edu/pgrams/OrthoFinder/orthofinder -f ../data/20-Apul-Orthofinder/proteins/ -t 20 -o ../output/20-Apul-Orthofinder/
```