---
title: "30.00-Apul-transcriptome-GOslims"
author: "Sam White"
date: "2025-06-17"
output: 
  github_document:
    toc: true
    number_sections: true
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
bibliography: references.bib
---

```{r setup, include=FALSE}
library(knitr)
library(dplyr)
library(readr)
library(Biostrings)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```


# VARIABLES

```{r variables}
# PROGRAMS
blastdbs_dir <- file.path("..", "..", "M-multi-species", "data", "blastdbs")
programs_dir <- file.path("", "home", "shared")
bedtools_dir <- file.path(programs_dir, "bedtools-v2.30.0", "bin")
blast_dir <- file.path(programs_dir, "ncbi-blast-2.15.0+", "bin")
diamond <- file.path(programs_dir, "diamond-2.1.8")
fastaFromBed <- file.path(bedtools_dir, "fastaFromBed")
samtools_dir <- file.path(programs_dir, "samtools-1.12")
samtools <- file.path(samtools_dir, "samtools")

# FILES
count_matrix <- "../output/07-Apul-Hisat/Apul-gene_count_matrix.csv"
diamond_db <- "20250618-diamond"
diamond_output <- file.path("..", "output", "Apulchra-expressed-genes.blastx.outfmt6")
genome_fasta <- file.path("..", "data", "Apulcra-genome.fa")
genes_fasta <- file.path("..", "output", "30.00-Apul-transcriptome-GOslims", "Apulchra-genes.fasta")
genes_fasta_index <- file.path("..", "output", "30.00-Apul-transcriptome-GOslims", "Apulchra-genes.fasta.fai")
genes_subset_fasta <- file.path("..", "output", "30.00-Apul-transcriptome-GOslims", "Apulchra-subset-genes.fasta")
genes_subset_fasta_index <- file.path("..", "output", "30.00-Apul-transcriptome-GOslims", "Apulchra-subset-genes.fasta.fai")
genes_bed <- file.path("..", "output", "30.00-Apul-transcriptome-GOslims", "Apulchra-genes.bed")
og_genome_gff <- file.path("..", "data", "Apulcra-genome.gff")

# THREADS
threads <- "40"


# FORMATTING
line <- "-----------------------------------------------"

# Export these as environment variables for bash chunks.
Sys.setenv(
  blastdbs_dir = blastdbs_dir,
  count_matrix = count_matrix,
  diamond = diamond,
  diamond_db = diamond_db,
  diamond_output = diamond_output,
  fastaFromBed = fastaFromBed,
  genes_bed = genes_bed,
  genes_fasta = genes_fasta,
  genes_fasta_index = genes_fasta_index,
  genes_subset_fasta = genes_subset_fasta,
  genes_subset_fasta_index = genes_subset_fasta_index,
  genome_fasta = genome_fasta,
  og_genome_gff = og_genome_gff,
  og_transcripts_gtf = og_transcripts_gtf,
  line = line,
  samtools = samtools,
  threads = threads,
  transcripts_fasta = transcripts_fasta,
  transcripts_gtf = transcripts_gtf
)
```

# Extract genes as FastA

## Extract genes from GFF
```{r extract-genes-from-gff}

# Read GFF, skipping comment lines
genome_gff <- readr::read_tsv(
  og_genome_gff,
  comment = "#",
  col_names = c("seqid", "source", "type", "start", "end", "score", "strand", "phase", "attributes")
)

# Filter for gene features
genes <- as.data.frame(genome_gff) %>%
  filter(type == "gene") %>%
  mutate(
    chrom = seqid,
    start = start - 1,  # BED is 0-based
    end = end,
    gene_id = sub("ID=([^;]+);?.*", "\\1", attributes)
  ) %>%
  select(chrom, start, end, gene_id)

str(genes)
```

### Write genes to BED
```{r write-genes-to-bed}
# Write to BED
write_tsv(genes, genes_bed, col_names = FALSE)
```

## Create genes FastA

```{r create-genes-FastA, engine='bash'}

"${fastaFromBed}" -fi "${genome_fasta}" -bed "${genes_bed}" -nameOnly > "${genes_fasta}"

# Create FastA index
"${samtools}" faidx "${genes_fasta}"

head "${genes_fasta}"

echo ""
echo ""

head "${genes_fasta_index}"
```
# Subset Expressed Genes

Only those with at least _one_ read in each sample

### Peek at count matrix

```{r check-count-matrix, engine=bash}
head "${count_matrix}" | column -t -s","

echo ""
echo ""

wc -l "${count_matrix}"
```


## Import count matrix
```{r import-count-matrix}

# Read the data into a data frame
count_matrix_df <- read.csv(count_matrix, header = TRUE)

str(count_matrix_df)
```

## Only genes with at least one read per sample
```{r subset-count-matrix}
# Filter rows where all values are greater than 0
filtered_count_matrix_df <- count_matrix_df[apply(count_matrix_df > 0, 1, all), ]


str(filtered_count_matrix_df)
```

## Subset genes FastA

Only expressed genes

```{r subset-genes-FastA}
# Get the row names (gene_ids) of the filtered data frame
filtered_gene_ids <- filtered_count_matrix_df$gene_id

fasta <- readDNAStringSet(genes_fasta)

subset_fasta <- fasta[names(fasta) %in% filtered_gene_ids]

```

```{r write-genes-subset-fasta}
writeXStringSet(subset_fasta, genes_subset_fasta)
```

### Peek at genes subset FastA
```{r check-genes-subset-fasta, engine=bash}
head "${genes_subset_fasta}"

echo ""
grep "^>" --count "${genes_subset_fasta}"
```

# BLASTx

## Download SwissProt
```{r download-sp, engine=bash}
cd "${blastdbs_dir}"

curl -O https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz

mv uniprot_sprot.fasta.gz 20250618-uniprot_sprot.fasta.gz

gunzip --keep 20250618-uniprot_sprot.fasta.gz
```

## Create BLASTdb
```{r create-BLASTdb, engine=bash}
cd "${blastdbs_dir}"

"${diamond}" makedb \
--in 20250618-uniprot_sprot.fasta.gz \
--db "${diamond_db}" \
--quiet \
--threads "${threads}"
```

## Run DIAMOND BLASTx
```{r diamond-blast, engine=bash}
"${diamond}" blastx \
--db "${blastdbs_dir}"/"${diamond_db}.dmnd" \
--query "${genes_subset_fasta}" \
--out "${diamond_output}" \
--outfmt 6  \
--sensitive \
--evalue 1e-10 \
--max-target-seqs 1 \
--block-size 15.0 \
--index-chunks 4 \
--threads "${threads}" \
2> ../output/30.00-Apul-transcriptome-GOslims/diamond-blastx.log

head "${diamond_output}"

echo ""

wc -l "${diamond_output}"
```