03.2-Ptuh-lncRNA-summary
================
Kathleen Durkin
2025-05-05

- <a href="#001-install-and-load-packages"
  id="toc-001-install-and-load-packages">0.0.1 Install and load
  packages</a>
- <a href="#1-unnormalized-data" id="toc-1-unnormalized-data">1
  Unnormalized data</a>
  - <a href="#11-load-count-data" id="toc-11-load-count-data">1.1 Load count
    data</a>
  - <a href="#12-count-data-munging" id="toc-12-count-data-munging">1.2
    Count data munging</a>
  - <a href="#13-expression-levels" id="toc-13-expression-levels">1.3
    Expression levels</a>
  - <a href="#14-transcript-counts" id="toc-14-transcript-counts">1.4
    Transcript counts</a>
- <a href="#2-normalized-counts" id="toc-2-normalized-counts">2 Normalized
  counts</a>
  - <a href="#21-normalize-counts-with-deseq2"
    id="toc-21-normalize-counts-with-deseq2">2.1 Normalize counts with
    DESeq2</a>
    - <a href="#211-plot-unnormalized-data"
      id="toc-211-plot-unnormalized-data">2.1.1 Plot unnormalized data</a>
    - <a href="#212-metadata" id="toc-212-metadata">2.1.2 Metadata</a>
    - <a href="#213-deseq-object" id="toc-213-deseq-object">2.1.3 DESeq
      object</a>
  - <a href="#22-plot-normalized-data" id="toc-22-plot-normalized-data">2.2
    Plot normalized data</a>
  - <a href="#23-plot-variance-stabilized-data"
    id="toc-23-plot-variance-stabilized-data">2.3 Plot variance stabilized
    data</a>
  - <a href="#24-normalized-expression-levels"
    id="toc-24-normalized-expression-levels">2.4 Normalized expression
    levels</a>
  - <a href="#25-normalized-transcript-counts"
    id="toc-25-normalized-transcript-counts">2.5 Normalized transcript
    counts</a>
  - <a href="#26-pca-of-variance-stabilized-data"
    id="toc-26-pca-of-variance-stabilized-data">2.6 PCA of variance
    stabilized data</a>
  - <a href="#27-sample-clustering" id="toc-27-sample-clustering">2.7 Sample
    clustering</a>
  - <a href="#28-heatmaps" id="toc-28-heatmaps">2.8 Heatmaps</a>

Re-ran 7/29/25 with updated lncRNA counts matrix

lncRNA expression summary for *Pocillopora tuahiniensis* lncRNA,
isolated in `18-Ptuh-lncRNA-matrix`

### 0.0.1 Install and load packages

``` r
library(tidyverse)
library(ggplot2)
library(reshape2)
library(magrittr)
library(DESeq2)
library(RColorBrewer)
library(pheatmap)
```

# 1 Unnormalized data

## 1.1 Load count data

``` r
# Read in counts data.
Ptuh_counts_lncRNA_OG <- read.table("../output/18-Ptuh-lncRNA-matrix/Ptuh-lncRNA-counts.txt", header=TRUE, skip=1) 
head(Ptuh_counts_lncRNA_OG)
```

          Geneid                                    Chr  Start    End Strand Length
    1 lncRNA_001 Pocillopora_meandrina_HIv1___Sc0000000 130008 130942      +    935
    2 lncRNA_002 Pocillopora_meandrina_HIv1___Sc0000000 164396 165221      +    826
    3 lncRNA_003 Pocillopora_meandrina_HIv1___Sc0000000 164598 165221      +    624
    4 lncRNA_004 Pocillopora_meandrina_HIv1___Sc0000000 168918 182502      +  13585
    5 lncRNA_005 Pocillopora_meandrina_HIv1___Sc0000000 245810 248612      +   2803
    6 lncRNA_006 Pocillopora_meandrina_HIv1___Sc0000000 282744 283165      +    422
      ...data.18.Ptuh.lncRNA.matrix.RNA.POC.47.sorted.bam
    1                                                 196
    2                                                  57
    3                                                   0
    4                                                 222
    5                                                   0
    6                                                 159
      ...data.18.Ptuh.lncRNA.matrix.RNA.POC.48.sorted.bam
    1                                                 319
    2                                                   0
    3                                                   0
    4                                                 362
    5                                                   0
    6                                                 158
      ...data.18.Ptuh.lncRNA.matrix.RNA.POC.50.sorted.bam
    1                                                 153
    2                                                  48
    3                                                   0
    4                                                 307
    5                                                 386
    6                                                 280
      ...data.18.Ptuh.lncRNA.matrix.RNA.POC.53.sorted.bam
    1                                                 274
    2                                                  16
    3                                                   0
    4                                                 709
    5                                                   0
    6                                                 345
      ...data.18.Ptuh.lncRNA.matrix.RNA.POC.57.sorted.bam
    1                                                 242
    2                                                  17
    3                                                   0
    4                                                 350
    5                                                  12
    6                                                 140

## 1.2 Count data munging

``` r
# We need to modify this data frame so that the row names are actually row names, instead of comprising the first column
Ptuh_counts_lncRNA <- Ptuh_counts_lncRNA_OG %>%
  column_to_rownames(var = "Geneid")

# Additional formatting
# Remove genomic coordinate info
Ptuh_counts_lncRNA <- Ptuh_counts_lncRNA %>% select(-Chr, -Start, -End, -Strand, -Length)

# Remove all transcripts with 5 or fewer counts in all samples
Ptuh_counts_lncRNA <- Ptuh_counts_lncRNA[!apply(Ptuh_counts_lncRNA, 1, function(row) all(row < 6)), ]

# Remove the superfluous portions of the column names, to leave just the sample names
colnames(Ptuh_counts_lncRNA) <- sub("...data.18.Ptuh.lncRNA.matrix.RNA.", "", colnames(Ptuh_counts_lncRNA))
colnames(Ptuh_counts_lncRNA) <- sub(".sorted.bam", "", colnames(Ptuh_counts_lncRNA))

# Reorder the columns into alphabetical order (to make it easier to create an associated metadata spreadsheet)
Ptuh_counts_lncRNA <- Ptuh_counts_lncRNA[, order(colnames(Ptuh_counts_lncRNA))]

write.table(Ptuh_counts_lncRNA, file = "../output/03.2-Ptuh-lncRNA-summary/Ptuh_lncRNA_counts_formatted.txt", row.names = TRUE, col.names = TRUE)

Ptuh_sample_names <- names(Ptuh_counts_lncRNA)

head(Ptuh_counts_lncRNA)
```

               POC.47 POC.48 POC.50 POC.53 POC.57
    lncRNA_001    196    319    153    274    242
    lncRNA_002     57      0     48     16     17
    lncRNA_004    222    362    307    709    350
    lncRNA_005      0      0    386      0     12
    lncRNA_006    159    158    280    345    140
    lncRNA_007    513     47    951     63    110

``` r
Ptuh_sample_names
```

    [1] "POC.47" "POC.48" "POC.50" "POC.53" "POC.57"

## 1.3 Expression levels

Plot histograms of the expression levels in each sample

``` r
# Melt the count matrix into long format
Ptuh_counts_melted <- melt(Ptuh_counts_lncRNA, variable.name = "sample", value.name = "counts")

# Plot the expression level histograms for each sample
ggplot(Ptuh_counts_melted, aes(x = counts)) +
  geom_histogram(binwidth = 1, fill = "#1E2761", color = "black") +
  scale_x_log10() +  # Optional: Log-transform the x-axis for better visualization
  facet_wrap(~sample, scales = "free_y") +
  labs(title = "lncRNA Expression Level Histogram for Each Sample",
       x = "Expression Level (Counts)",
       y = "Frequency") +
  theme_minimal()
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/expression-level-histograms-1.png)<!-- -->

## 1.4 Transcript counts

First let’s check the total number of transcripts in each sample – keep
in mind this expression data has *not* been normalized yet, so there may
be different totals for each sample

``` r
# Calculate the total number of transcripts for each sample
total_transcripts <- colSums(Ptuh_counts_lncRNA)

# Create a data frame for plotting
total_transcripts_df <- data.frame(sample = names(total_transcripts),
                                   totals = total_transcripts)

# Plot the total number of transcripts for each sample
ggplot(total_transcripts_df, aes(x = sample, y = totals)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Transcripts per Sample",
       x = "Sample",
       y = "Total Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/transcript-counts-plot-1.png)<!-- -->

Now let’s check the number of unique transcripts in each sample – that
is, how many lncRNAs are expressed in each sample? This should be pretty
much the same across samples, even without normalization.

``` r
# Calculate the number of unique transcripts (non-zero counts) for each sample
unique_transcripts <- colSums(Ptuh_counts_lncRNA > 0)

# Create a data frame for plotting
unique_transcripts_df <- data.frame(sample = names(unique_transcripts),
                                    uniques = unique_transcripts)

# Plot the total number of unique transcripts for each sample
ggplot(unique_transcripts_df, aes(x = sample, y = uniques)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Unique Expressed Transcripts per Sample",
       x = "Sample",
       y = "Unique Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/total-unique-transcripts-plot-1.png)<!-- -->

Some variation

# 2 Normalized counts

## 2.1 Normalize counts with DESeq2

### 2.1.1 Plot unnormalized data

``` r
Ptuh_counts_lncRNA %>%
  pivot_longer( cols = everything(), names_to = "sample", values_to = "count") %>%
  ggplot(., aes(x = sample, y = count)) +
  geom_violin() + 
  geom_point(alpha = 0.2) +
  theme_minimal() +
  labs(title = "Unnormalized transcript counts",
       x = "Sample",
       y = "count")
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/plot-unnormalized-RNA-1.png)<!-- -->

### 2.1.2 Metadata

DESeq2 requires a metadata data frame as input. I don’t have sample
metadata though so, since we’re just doing DESeq2 for normalization
purposes (not analysis purposes), I’m just going to create a dummy sheet

``` r
Ptuh_sample_names <- colnames(Ptuh_counts_lncRNA)

Ptuh_metadata_lncRNA <- data.frame(Sample = Ptuh_sample_names,
                            Species = rep("P.tuahiniensis", 5))
rownames(Ptuh_metadata_lncRNA) <- Ptuh_sample_names

head(Ptuh_metadata_lncRNA)
```

           Sample        Species
    POC.47 POC.47 P.tuahiniensis
    POC.48 POC.48 P.tuahiniensis
    POC.50 POC.50 P.tuahiniensis
    POC.53 POC.53 P.tuahiniensis
    POC.57 POC.57 P.tuahiniensis

### 2.1.3 DESeq object

``` r
# Calculate DESeq object
dds_Ptuh_lncRNA <- DESeqDataSetFromMatrix(countData = Ptuh_counts_lncRNA,
                              colData = Ptuh_metadata_lncRNA,
                              design = ~ 1) 

# Run differential expression analysis 
# (Note that this DESeq() function runs all necessary steps, including data normalization, 
# estimating size factors, estimating dispersions, gene-wise dispersion estimates, mean-dispersion 
# relationship, final dispersion estimates, fitting model, and testing)
# Using design = ~1 because we don't have treatment groups

dds_Ptuh_lncRNA <- DESeq(dds_Ptuh_lncRNA)
```

It’s worth noting here that I’m actually going to be doing two different
types of transformation on the counts data, which serve different
purposes.

- First is **normalizing** the transcript counts, which adjusts for
  differences in library size or sequencing depth, but retains
  count-like properties. Normalized counts are most useful for things
  like visualizing expression levels and differential expression
  analysis.

- Second is **variance stabilizing** the counts data, which aims to make
  the variance of the transformed data approximately independent of the
  mean, reducing heteroscedasticity (the relationship between variance
  and mean) and “smoothing” out the variance at low counts. Notably, the
  transformed data is *no longer on the original count scale*. The
  transformation makes the variance roughly constant across the range of
  counts, which makes it easier to interpret patterns in the data
  visually. Variance stabilized data is most useful for exploratory data
  analysis, like PCA, clustering, and heatmaps, and is also the
  transformation we’ll want to use before WGCNA.

``` r
# extract normalized counts
# (normalization is automatically performed by deseq2)
Ptuh_counts_lncRNA_norm <- counts(dds_Ptuh_lncRNA, normalized=TRUE) %>% data.frame()

write.table(Ptuh_counts_lncRNA_norm, file = "../output/03.2-Ptuh-lncRNA-summary/Ptuh_counts_lncRNA_normalized.txt", sep = "\t", row.names = TRUE, col.names = TRUE, quote = FALSE)


# variance stabilized data
vsd_Ptuh_lncRNA <- varianceStabilizingTransformation(dds_Ptuh_lncRNA, blind=TRUE)
wpn_vsd_Ptuh_lncRNA <- getVarianceStabilizedData(dds_Ptuh_lncRNA)
rv_wpn_Ptuh_lncRNA <- rowVars(wpn_vsd_Ptuh_lncRNA, useNames=TRUE)

Ptuh_counts_lncRNA_vsd <- data.frame(wpn_vsd_Ptuh_lncRNA)
write.table(Ptuh_counts_lncRNA_vsd, file = "../output/03.2-Ptuh-lncRNA-summary/Ptuh_counts_lncRNA_variancestabilized.txt", sep = "\t", row.names = TRUE, col.names = TRUE,quote = FALSE)

q75_wpn_Ptuh_lncRNA <- quantile(rowVars(wpn_vsd_Ptuh_lncRNA, useNames=TRUE), .75)  # 75th quantile variability
Ptuh_counts_lncRNA_vsd_q75 <- wpn_vsd_Ptuh_lncRNA[ rv_wpn_Ptuh_lncRNA > q75_wpn_Ptuh_lncRNA, ] %>% data.frame # filter to retain only the most variable lncRNAs
write.table(Ptuh_counts_lncRNA_vsd_q75, file = "../output/03.2-Ptuh-lncRNA-summary/Ptuh_counts_lncRNA_variancestabilized_q75.txt", sep = "\t", row.names = TRUE, col.names = TRUE,quote = FALSE)

q95_wpn_Ptuh_lncRNA <- quantile(rowVars(wpn_vsd_Ptuh_lncRNA, useNames=TRUE), .95)  # 95th quantile variability
Ptuh_counts_lncRNA_vsd_q95 <- wpn_vsd_Ptuh_lncRNA[ rv_wpn_Ptuh_lncRNA > q95_wpn_Ptuh_lncRNA, ] %>% data.frame # filter to retain only the most variable lncRNAs
write.table(Ptuh_counts_lncRNA_vsd_q95, file = "../output/03.2-Ptuh-lncRNA-summary/Ptuh_counts_lncRNA_variancestabilized_q95.txt", sep = "\t", row.names = TRUE, col.names = TRUE,quote = FALSE)
```

## 2.2 Plot normalized data

``` r
Ptuh_counts_lncRNA_norm_long <- Ptuh_counts_lncRNA_norm %>%
  mutate(
    Gene_id = row.names(Ptuh_counts_lncRNA_norm)
  ) %>%
  pivot_longer(-Gene_id)

Ptuh_counts_lncRNA_norm_long %>%
  ggplot(., aes(x = name, y = value)) +
  geom_violin() +
  geom_point() +
  theme_bw() +
  theme(
    axis.text.x = element_text( angle = 90)
  ) +
  ylim(0, NA) +
  labs(
    title = "Normalized Expression",
    x = "Sample",
    y = "Normalized counts"
  )
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/plot-normalized-RNA-1.png)<!-- -->

## 2.3 Plot variance stabilized data

``` r
Ptuh_counts_lncRNA_vsd_long <- Ptuh_counts_lncRNA_vsd %>%
  mutate(
    Gene_id = row.names(Ptuh_counts_lncRNA_vsd)
  ) %>%
  pivot_longer(-Gene_id)

Ptuh_counts_lncRNA_vsd_long %>%
  ggplot(., aes(x = name, y = value)) +
  geom_violin() +
  geom_point() +
  theme_bw() +
  theme(
    axis.text.x = element_text( angle = 90)
  ) +
  ylim(0, NA) +
  labs(
    title = "Variance Stabilized Expression",
    x = "Sample",
    y = "Variance stabilized data"
  )
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/plot-vsd-RNA-1.png)<!-- -->

## 2.4 Normalized expression levels

Plot histograms of the normalized expression levels in each sample

``` r
# Melt the count matrix into long format
Ptuh_counts_norm_melted <- melt(Ptuh_counts_lncRNA_norm, variable.name = "sample", value.name = "counts")

# Plot the expression level histograms for each sample
ggplot(Ptuh_counts_norm_melted, aes(x = counts)) +
  geom_histogram(binwidth = 1, fill = "#1E2761", color = "black") +
  scale_x_log10() +  # Optional: Log-transform the x-axis for better visualization
  facet_wrap(~sample, scales = "free_y") +
  labs(title = "lncRNA Expression Level Histogram for Each Sample",
       x = "Expression Level (Counts)",
       y = "Frequency") +
  theme_minimal()
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/norm-expression-level-histograms-1.png)<!-- -->

## 2.5 Normalized transcript counts

Check the total number of transcripts in each sample – now that we’ve
normalized the data these totals should be similar

``` r
# Calculate the total number of transcripts for each sample
total_transcripts_norm <- colSums(Ptuh_counts_lncRNA_norm)

# Create a data frame for plotting
total_transcripts_norm_df <- data.frame(sample = names(total_transcripts_norm),
                                   totals = total_transcripts_norm)

# Plot the total number of transcripts for each sample
ggplot(total_transcripts_norm_df, aes(x = sample, y = totals)) +
  geom_bar(stat = "identity", fill = "#1E2761", color = "black") +
  labs(title = "Total Number of Transcripts per Sample",
       x = "Sample",
       y = "Total Transcripts") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/norm-transcript-counts-plot-1.png)<!-- -->

## 2.6 PCA of variance stabilized data

``` r
plotPCA(vsd_Ptuh_lncRNA, intgroup="Sample")
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/PCA-1.png)<!-- -->

As in Apul and Peve, there is a fair bit of divergence betweem samples.
However, unlike the other two species, this variation doesn’t seem to be
driven by differences in normalized transcript counts. The two samples
with highest normalized counts are POC.47 and POC.48. However, there
POC.47 is only one of the samples that pulls out on the PCC, and POC.48
clusters with another sample with lower counts (POC.53). We also se in
the sample clustering (below) that POC.47is most similar to POC.57, and
that both are notably different from the other 3 samples. This is
despite POC.57 having very similar normalized transcript counts to
POC.53 and POC.50. This all suggest that in Ptuh, unlike the other two
species, the divergent lncRNA profiles seem driven by composition, not
just total counts (overall expression level).

## 2.7 Sample clustering

``` r
sample_dists <- dist(t(assay(vsd_Ptuh_lncRNA)))
pheatmap(as.matrix(sample_dists), clustering_distance_rows = "euclidean", 
         clustering_distance_cols = "euclidean", main="Sample Clustering")
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/sample-clustering-1.png)<!-- -->

## 2.8 Heatmaps

Of most variable variance stabilized lncRNAs

``` r
# 75th quantile
heat_colors <- rev(brewer.pal(12, "RdYlBu"))
pheatmap(Ptuh_counts_lncRNA_vsd_q75, 
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         show_rownames = TRUE,
         show_colnames = TRUE,
         color = heat_colors,
         scale="row")
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/heatmpas-1.png)<!-- -->

``` r
# 95th quantile
pheatmap(Ptuh_counts_lncRNA_vsd_q95, 
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         show_rownames = TRUE,
         show_colnames = TRUE,
         color = heat_colors,
         scale="row")
```

![](03.2-Ptuh-lncRNA-summary_files/figure-gfm/heatmpas-2.png)<!-- -->
